{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ChronoSeq Overview\n",
    "\n",
    "### ChronoSeq Device Simplified Operation Principle\n",
    "\n",
    "|First Timepoint Injection|Second Timepoint Injection|\n",
    "|-|-|\n",
    "|<img src=\"img/Simplified%20Figure%20T1.jpg\">|<img src=\"img/Simplified%20Figure%20T2.jpg\">|\n",
    "\n",
    "|Third Timepoint Injection|Fourth Timepoint Injection|\n",
    "|-|-|\n",
    "|<img src=\"img/Simplified%20Figure%20T3.jpg\">|<img src=\"img/Simplified%20Figure%20T4.jpg\">|\n",
    "\n",
    "\n",
    "+ The device sequentially co-injects time-tagged beads and cells into a droplet-generating microfluidic chip at regular time intervals.\n",
    "+ These beads are stored in separate reservoirs (e.g., 4 reservoirs for 4 time points for illustration purpose), with each reservoir containing beads with a unique time-tag.\n",
    "+ For each timepoint, the Cell Sample Valve remains open while the corresponding Timepoint Valve (orange for T1, yellow for T2, Hot Pink for T3, and Olive Green for T4) is opened. \n",
    "    + This allows for the co-injection of time-specific beads with cells, resulting in libraries barcoded with distinct time-tags. \n",
    "+ This process enables precise temporal labeling of cells across different timepoints. Our system currently supports 12 unique time-tags with 7-minute intervals and 1-minute sampling duration."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Simplified Video Explaining Operation\n",
    "+ Execute the cell below to watch the video."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<iframe width=\"800\" height=\"600\" src=\"https://www.youtube.com/embed/pb1Ijufe26k?si=u0dQUczlU5yyl1fT\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" referrerpolicy=\"strict-origin-when-cross-origin\" allowfullscreen></iframe>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%HTML\n",
    "<iframe width=\"800\" height=\"600\" src=\"https://www.youtube.com/embed/pb1Ijufe26k?si=u0dQUczlU5yyl1fT\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" referrerpolicy=\"strict-origin-when-cross-origin\" allowfullscreen></iframe>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "| <h2>ChronoSeq V4 Workflow</h2> |\n",
    "|-|\n",
    "|<img src=\"img/ChronoSeq%20workflow.png\">|\n",
    "\n",
    "### The figure above illustrates the Library preparation workflow.\n",
    "1. **Bead structure:** Modified from the [Dropseq](https://www.cell.com/fulltext/S0092-8674(15)00549-8) beads, our beads contain cell barcodes and unique molecular identifiers (UMIs). Our beads also have an additional Time-Tag common for each Bead Reservoir. \n",
    "2. **Droplet generation:** Cells and time-tagged beads are co-injected into a microfluidic chip that uses [inertial ordering](https://doi.org/10.1039/C7LC01284E) to co-encapsulate one bead with one cell in each droplet. \n",
    "3. and 4. **Cell lysis and molecular capture:** Upon droplet formation, cells are lysed, releasing their mRNA. These are captured by the PolyT region on the beads. \n",
    "5. **Sample processing:** Droplets are collected and kept on ice until all time points are sampled. Droplets are then broken, and beads are combined, followed by a combined reverse transcription reaction. \n",
    "6. **Library preparation.** A RNA-Seq library is prepared. This is then followed by tagmentation and sequencing adapter addition.\n",
    "7. and 8. **Sequencing and data analysis:** After sequencing, a Digital data matrix is created. Time-tags are assigned to each cell barcode, allowing temporal analysis of gene expression.\n",
    "\n",
    "### Design of equipment.\n",
    "\n",
    "|ChronoSeq Device Schematic|\n",
    "|-|\n",
    "|<img src=\"Main%20Figure.jpg\">|\n",
    "\n",
    "Our device design addresses several challenges not present in the original Dropseq system (See Figure Below). First, extended cell viability. We maintain cell suspension viability for experiments lasting 2 hours or longer, compared to typical 30-minute Drop-seq runs. Second, cross-contamination prevention. We implemented a washing system to remove unused cells or beads between samples. Third, sample isolation. We designed a system to prevent mixing wash fluid with droplets. To address these challenges, our device operates in two main phases during each injection cycle. 1. Flushing phase that clears fluid lines of residual beads or cells from previous injections. 2. Injection phase that precisely introduces cells and time-tagged beads into the microfluidic chip. This integrated system enables controlled, sequential introduction of cells and time-tagged beads, capturing gene expression dynamics across multiple time points in a single experiment.\n",
    "\n",
    "| <h2>ChronoSeq vs Dropseq comparison</h2> |\n",
    "|-|\n",
    "|<img src=\"img/Dropseq%20vs%20ChronoSeq%20Comparison.jpeg\">|\n",
    "\n",
    "### The Figure above compares the ChronoSeq device (A) with Dropseq device (B). \n",
    "The ChronoSeq device is controlled via serial connections to a computer, which coordinates the actions of its various components, while the Dropseq device works by setting flow rates manually on syringe pumps. Some key features distinguish the ChronoSeq device:\n",
    "1. **Bead handling:** ChronoSeq incorporates an automated vortex mixer to keep the beads evenly suspended before injection. Dropseq uses a magnetic stirrer for bead suspension, which leads to high dead volumes and complicates automation for multiple time-tags. ChronoSeq allows for the easy addition of more time-tags in the future if necessary. Additionally, loading and unloading the beads is simplified; we can directly screw a new 50 mL tube containing beads into the reservoir. The time-tagged bead reservoirs are connected to timepoint valves, which facilitate the selection of the appropriate time-tagged beads for injection into the microfluidic chip. \n",
    "2. **Cell culture system:** In the Dropseq device, cells are directly loaded into a syringe for injection, typically running for 15 to 30 minutes. However, our goal is to maintain cell viability for at least 2 hours. Leaving cells at room temperature without growth media and in the absence of 5% CO<sub>2</sub> can lead to excessive cell death and unwanted perturbations. To address this, we have integrated a cell culture system directly into the ChronoSeq device, featuring a water bath to maintain cells at 37&deg;C, along with a 5% CO<sub>2</sub> gas cylinder that serves as a pressure source and aids in cell culture. Furthermore, we can add additional tubes to culture multiple cell suspensions separately and use valves to select which cell suspension is injected into the microfluidic chip. \n",
    "3. **Droplet collection:** In the Dropseq device, droplets are collected directly in a 50 mL tube kept on ice. However, in our system, it is crucial to prevent the beads and cells from previous injections from mixing with those from new injections. To achieve this, we developed a washing system to clear the fluid lines of residual beads and cells. We also need to ensure that washing liquids and unused cell suspensions do not mix with the collected droplets. To solve this issue, we designed a robotic system that facilitates droplet collection and bead recovery. This robot automatically moves the outlet tubing to the correct tube, preventing the mixing of droplets with washing fluids. Additionally, it can recover unused beads from the fluid lines, minimizing waste and reducing experimental costs. Like the Dropseq device, the droplets in our system are also kept on ice using ice boxes. \n",
    "4. **Flow control:** In the Dropseq device, precise speed control is essential for droplet formation. Speeds that are too high can inhibit droplet formation, while low speeds can negatively impact throughput. To achieve this precise control, Dropseq employs syringe pumps, which allow direct manipulation of the syringe's squeezing rate. In our case, we require high flow rates for flushing and low flow rates for droplet formation. Low flow rates like Dropseq flow rates, are insufficient for quickly washing unused beads and cells from the device. Therefore, we implemented a compressed air pressure controller and flow sensors for each of the bead, cell, and oil channels. The compressed air controller enables rapid changes in flow rate, while the flow sensors provide the precise control needed for high-quality droplet formation during low flow rate operation.\n",
    "\n",
    "### Microfluidic Chip Design\n",
    "|ChronoSeq Microfluidic Chip Design V14|\n",
    "|-|\n",
    "|<img src=\"img/Chrono-Seq%20Chip.jpeg\">|\n",
    "\n",
    "The primary function of the microfluidic chip is to co-encapsulate cells with time-tagged beads using a [flow-focusing junction](https://doi.org/10.1002/elps.200500173) at the center of the chip. While we initially considered using the Dropseq microfluidic chip for this purpose, several limitations made it unsuitable for our application: \n",
    "1. **Ordering mechanism:** Dropseq relies on random (Poisson) ordering for co-encapsulation of cells and beads. This random ordering makes it difficult to increase the number of cells collected per timepoint without also increasing droplets containing multiple cells or beads. The ChronoSeq microfluidic chip uses an optimized version of an [existing inertial ordering design](https://doi.org/10.1039/C7LC01284E), to increase cell capture through deterministic ordering of beads into droplets. This deterministic ordering allows for capturing more cells by increasing the number of droplets containing exactly one bead and one cell. \n",
    "2. **Channel design:** Both the Dropseq chip and existing inertial ordering designs use narrow flow resistance channels, which are prone to blockage during the flushing phase of the ChronoSeq device. To reduce blockage risk, we widened the channels in our chip design. To maintain the same flow resistance with wider channels, we increased the length of the channels to compensate (See Figure Below).\n",
    "3. **Quality Control:** We implemented a rigorous quality control protocol: compressed air testing to check for delamination and eliminate leaking chips, simulating the high pressures used during the flushing phase; visual inspection to ensure the absence of particles before connecting to the device. These design and quality control measures enable the chips to withstand higher pressures and cyclical pressurization during the flushing phase. The optimized chip design is crucial for reliable, high-throughput, time-resolved single-cell sequencing. The improvements in channel geometry and manufacturing process contribute to reduced clogging, increased cell capture efficiency, overall system robustness. These features are essential for maintaining consistent performance across multiple time points and ensuring high-quality data collection throughout extended experimental runs.\n",
    "\n",
    "|<h2>ChronoSeq vs Dropseq chip comparison</h2>|\n",
    "|-|\n",
    "|<img src=\"img/Drop-Seq%20Vs%20Chrono-Seq%20Chip.png\">|\n",
    "\n",
    "#### The ChronoSeq chip has three key modifications compared to the Drop-seq chip.\n",
    "1. The Droplet Mixer near the Drop-Seq outlet hole has been removed to reduce the likelihood of blockage.\n",
    "2. The width of the flow-resistors has been increased along with the length of the channels to keep the flow resistance identical between the two designs. \n",
    "3. The straight (Poisson) ordering inlet has been replaced with an inertial (deterministic) ordering spiral design (using an existing inertial ordering design as a template).\n",
    "\n",
    "### Creation of ChronoSeq V4 Beads for Time-Tagged RNA-Seq\n",
    "\n",
    "|<h3>Bead Modification Workflow for creating Time-Tagged Beads</h3>|\n",
    "|-|\n",
    "|<img src=\"img/ChronoSeq%20V5%20without%20Exonuclease%20Step.png\">|\n",
    "\n",
    "Sequences are added by using an oligo that binds to the existing PolyT region on Drop-seq beads. The 3’ end of the Drop-seq beads are then extended to the reverse compliment of the oligo using E. Coli DNA Polymerase I. Alkaline denaturation is used to make the double stranded DNA single-stranded. These beads are then resuspended in lysis buffer and can be used directly with our device.\n",
    "\n",
    "### Single-Cell Experiment to verify Quality of Data Generated Using ChronoSeq technology.\n",
    "\n",
    "|50:50 Human(K562) and Mouse(EL4) Cell Suspension|Human only (K562)| Mouse only (EL4)|\n",
    "|-|-|-|\n",
    "|<img src=\"img/SEQ10.svg\">|<img src=\"img/SEQ11.svg\">|<img src=\"img/SEQ12.svg\">|\n",
    "\n",
    "#### Species-Mixing Barnyard plots are for the same single-cell experiment with a 10min sampling interval and 1 minute sampling duration.\n",
    "3 samples were taken and barcoded with three separate Time-Tags. The first sample (left) was from a 50:50 Human(K562) and Mouse(EL4) Cell Suspension. The second sample (center) was from a Human only cell suspension. While the third sample (right) was from a Mouse only cell suspension. Each dot in the three plots represents a Unique Cell Barcode and the (X, Y) coordinates represent the number of Human and Mouse transcripts respectively.<br>\n",
    "\n",
    "From this data we can conclude:\n",
    "+ The time-tags can separate each injection uniquely, meaning there are almost no time-tags from the previous injection and no cross-contamination.\n",
    "+ Moreover, the time-tags capture only the cells co-injected with them, showing effective temporal barcoding of cells.\n",
    "+ Next, the data generated in the first time-tag has very few mixed cell barcodes from human and mouse, meaning human and mouse cells are unique captured with low noise, comparable to other technologies.\n",
    "\n",
    "### Bulk RNA-seq shows key time-delays and kinetic differences for gene activation\n",
    "\n",
    "|TNF&alpha; added at 15mins|TNF&alpha; added at 65mins|\n",
    "|-|-|\n",
    "|<img src=\"img/15minNFKBIA-ICAM1.svg\">|<img src=\"img/65minNFKBIA-ICAM1.svg\">|\n",
    "|<img src=\"img/15minNFKBIA-TNFRSF9.svg\">|<img src=\"img/65minNFKBIA-TNFRSF9.svg\">|\n",
    "|<img src=\"img/15minNFKBIA-IL8.svg\">|<img src=\"img/65minNFKBIA-IL8.svg\">|\n",
    "\n",
    "#### Bulk RNA-seq from K562 cells simulated with TNFα at 15 mins (left) and 65 mins (right) show key time-delays and kinetic differences for gene activation.\n",
    "X-axis is time in minutes and Y-axis is the log normalized fold change with respect to an unperturbed control for several representative genes. Because of the delayed addition of TNFα to the experiment on the right, the activation of these representative genes is delayed compared to the experiment on the left with a shifting pattern. However, both experiments exhibit diverse delay and rate of gene activation. For example, NFKBIA gets activated fastest, ICAM1 next and TNFRSF9 last. This delayed activation of TNFRSF9 indicates that it is not the initiator of the expression change cascade, providing valuable causal insights. Additionally, even genes activated simultaneously can display different activation rates; for example, NFKBIA shows a rapid, steep increase, whereas IL8 demonstrates a more gradual rise. These fine-grained temporal dynamics would be undetectable using conventional technologies, which typically have lower temporal resolution.  \n",
    "\n",
    "### Prior Art and References\n",
    "1. Abate, A., Li, X. & Demaree, B. Partitioning cells for high throughput single-cell sequencing. (2024).\n",
    "2. Abate, A. R., HALIBURTON, J., LAN, F. & SCIAMBI, A. R. Sequencing of nucleic acids via barcoding in discrete entities. (2016).\n",
    "3. Bar-Joseph, Z., Gitter, A. & Simon, I. Studying and modelling dynamic biological processes using time-series gene expression data. Nat Rev Genet 13, 552–564 (2012).\n",
    "4. Battich, N. et al. Sequencing metabolically labeled transcripts in single cells reveals mRNA turnover strategies. Science 367, 1151–1156 (2020).\n",
    "5. Buenrostro, J. D. et al. Single-cell chromatin accessibility reveals principles of regulatory variation. Nature 523, 486–490 (2015).\n",
    "6. Cao, J. et al. Joint profiling of chromatin accessibility and gene expression in thousands of single cells. Science 361, 1380–1385 (2018).\n",
    "7. Cao, J. et al. Comprehensive single-cell transcriptional profiling of a multicellular organism. Science 357, 661–667 (2017).\n",
    "8. Chang, C. & Shum, E. Determining 5’ transcript sequences. (2020).\n",
    "9. Chee, M. S. & Gunderson, K. Detection of nucleic acid amplification reactions using bead arrays. (2002).\n",
    "10. Chee, M. S., Gunderson, K. L. & Weiner, M. P. Macromolecule analysis employing nucleic acid encoding. (2024).\n",
    "11. Church, G. M. & Vigneault, F. High-Throughput Single Cell Barcoding. (2013).\n",
    "12. Clark, I. C. et al. Microfluidics-free single-cell genomics with templated emulsification. Nat Biotechnol 41, 1557–1566 (2023).\n",
    "13. Clark, S. J. et al. scNMT-seq enables joint profiling of chromatin accessibility DNA methylation and transcription in single cells. Nat Commun 9, 781 (2018).\n",
    "14. Coulon, A., Chow, C. C., Singer, R. H. & Larson, D. R. Eukaryotic transcriptional dynamics: from single molecules to cell populations. Nat Rev Genet 14, 572–584 (2013).\n",
    "15. Cusanovich, D. A. et al. A Single-Cell Atlas of In Vivo Mammalian Chromatin Accessibility. Cell 174, 1309-1324.e18 (2018).\n",
    "16. De Rop, F. V. et al. Systematic benchmarking of single-cell ATAC-sequencing protocols. Nat Biotechnol 42, 916–926 (2024).\n",
    "17. Delley, C. L. & Abate, A. R. Modular barcode beads for microfluidic single cell genomics. Sci Rep 11, 10857 (2021).\n",
    "18. Deng, Y. et al. Spatial profiling of chromatin accessibility in mouse and human tissues. Nature 609, 375–383 (2022).\n",
    "19. Dukler, N. et al. Nascent RNA sequencing reveals a dynamic global transcriptional response at genes and enhancers to the natural medicinal compound celastrol. Genome Res. 27, 1816–1829 (2017).\n",
    "20. Erhard, F. et al. scSLAM-seq reveals core features of transcription dynamics in single cells. Nature 571, 419–423 (2019).\n",
    "21. Erhard, F. et al. Time-resolved single-cell RNA-seq using metabolic RNA labelling. Nat Rev Methods Primers 2, 1–18 (2022).\n",
    "22. Fan, C., Fodor, S. P. A., Fu, G., Facer, G. R. & Wilhelmy, J. Massively parallel single cell analysis. (2017).\n",
    "23. Fan, C., Fodor, S. P. A., Fu, G., Facer, G. R. & Wilhelmy, J. Massively parallel single cell analysis. (2018).\n",
    "24. Fang, R. et al. Comprehensive analysis of single cell ATAC-seq data with SnapATAC. Nat Commun 12, 1337 (2021).\n",
    "25. Grunenwald, H. L., Caruccio, N., Jendrisak, J. & Dahl, G. Transposon end compositions and methods for modifying nucleic acids. (2010).\n",
    "26. Han, X. et al. Mapping the Mouse Cell Atlas by Microwell-Seq. Cell 172, 1091-1107.e17 (2018).\n",
    "27. Hao, S. & Baltimore, D. The stability of mRNA influences the temporal order of the induction of genes encoding inflammatory molecules. Nat Immunol 10, 281–288 (2009).\n",
    "28. He, S. et al. High-plex imaging of RNA and proteins at subcellular resolution in fixed tissue by spatial molecular imaging. Nat Biotechnol 40, 1794–1806 (2022).\n",
    "29. Hu, Y. et al. Simultaneous profiling of transcriptome and DNA methylome from a single cell. Genome Biol 17, 88 (2016).\n",
    "30. Hung, D., Ma, P. & Amemiya, H. High-throughput, droplet-based single cell rna sequencing. (2023).\n",
    "31. Iyer, V. R. et al. The Transcriptional Program in the Response of Human Fibroblasts to Serum. Science 283, 83–87 (1999).\n",
    "32. Jaitin, D. A. et al. Massively Parallel Single-Cell RNA-Seq for Marker-Free Decomposition of Tissues into Cell Types. Science 343, 776–779 (2014).\n",
    "33. Kaya-Okur, H. S. et al. CUT&Tag for efficient epigenomic profiling of small samples and single cells. Nat Commun 10, 1930 (2019).\n",
    "34. Klein, A. M. et al. Droplet Barcoding for Single-Cell Transcriptomics Applied to Embryonic Stem Cells. Cell 161, 1187–1201 (2015).\n",
    "35. Klemm, S. L., Shipony, Z. & Greenleaf, W. J. Chromatin accessibility and the regulatory epigenome. Nat Rev Genet 20, 207–220 (2019).\n",
    "36. Kleshchevnikov, V. et al. Cell2location maps fine-grained cell types in spatial transcriptomics. Nat Biotechnol 40, 661–671 (2022).\n",
    "37. Liu, H. et al. SLAM‐Drop‐seq reveals mRNA kinetic rates throughout the cell cycle. Molecular Systems Biology 19, e11427 (2023).\n",
    "38. Macosko, E. Z. et al. Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets. Cell 161, 1202–1214 (2015).\n",
    "39. Mahat, D. B. et al. Single-cell nascent RNA sequencing unveils coordinated global transcription. Nature 631, 216–223 (2024).\n",
    "40. Moon, H.-S. et al. Inertial-ordering-assisted droplet microfluidics for high-throughput single-cell RNA-sequencing. Lab Chip 18, 775–784 (2018).\n",
    "41. Morabito, S. et al. Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimer’s disease. Nat Genet 53, 1143–1155 (2021).\n",
    "42. Ngo, K. A. et al. Dissecting the Regulatory Strategies of NF-κB RelA Target Genes in the Inflammatory Response Reveals Differential Transactivation Logics. Cell Reports 30, 2758-2775.e6 (2020).\n",
    "43. Oliveira, M. F. et al. Characterization of immune cell populations in the tumor microenvironment of colorectal cancer using high definition spatial profiling. 10.1101/2024.06.04.597233 (2024).\n",
    "44. Ranek, J. S., Stanley, N. & Purvis, J. E. Integrating temporal single-cell gene expression modalities for trajectory inference and disease prediction. Genome Biology 23, 186 (2022).\n",
    "45. Rao, A., Barkley, D., França, G. S. & Yanai, I. Exploring tissue architecture using spatial transcriptomics. Nature 596, 211–220 (2021).\n",
    "46. Regev, A. et al. A droplet-based method and apparatus for composite single-cell nucleic acid analysis. (2016).\n",
    "47. Russell, A. J. C. et al. Slide-tags enables single-nucleus barcoding for multimodal spatial genomics. Nature 625, 101–109 (2024).\n",
    "48. Satpathy, A. T. et al. Massively parallel single-cell chromatin landscapes of human immune cell development and intratumoral T cell exhaustion. Nat Biotechnol 37, 925–936 (2019).\n",
    "49. Stickels, R. R. et al. Highly sensitive spatial transcriptomics at near-cellular resolution with Slide-seqV2. Nat Biotechnol 39, 313–319 (2021).\n",
    "50. Stoeckius, M. et al. Simultaneous epitope and transcriptome measurement in single cells. Nat Methods 14, 865–868 (2017).\n",
    "51. Stoeckius, M., Smibert, P. & Houck-Loomis, B. Methods and compositions for identifying or quantifying targets in a biological sample. (2018).\n",
    "52. Swanson, E. et al. Simultaneous trimodal single-cell measurement of transcripts, epitopes, and chromatin accessibility using TEA-seq. eLife 10, e63632 (2021).\n",
    "53. Tian, B., Nowak, D. E. & Brasier, A. R. A TNF-induced gene expression program under oscillatory NF-κB control. BMC Genomics 6, 137 (2005).\n",
    "54. Vandereyken, K., Sifrim, A., Thienpont, B. & Voet, T. Methods and applications for single-cell and spatial multi-omics. Nat Rev Genet 24, 494–515 (2023).\n",
    "55. Vigneault, F. & BRIGGS, A. W. Single cell bar-coding for antibody discovery. (2014).\n",
    "56. Wang, X., Lim, H. J. & Son, A. Characterization of denaturation and renaturation of DNA for DNA hybridization. Environ Health Toxicol 29, (2014).\n",
    "57. Wang, Y. et al. Dissolvable Polyacrylamide Beads for High‐Throughput Droplet DNA Barcoding. Advanced Science 7, 1903463 (2020).\n",
    "58. Ward, T., Faivre, M., Abkarian, M. & Stone, H. A. Microfluidic flow focusing: Drop size and scaling in pressure flow-rate-driven pumping. ELECTROPHORESIS 26, 3716–3724 (2005).\n",
    "59. Weitz, D. A., KLEIN, A. M., Akartuna, I., Mazutis, L. & Kirschner, M. W. Systems and methods for barcoding nucleic acids. (2015).\n",
    "60. Wissink, E. M., Vihervaara, A., Tippens, N. D. & Lis, J. T. Nascent RNA analyses: tracking transcription and its regulation. Nat Rev Genet 20, 705–723 (2019).\n",
    "61. Xie, Y. et al. Droplet-based single-cell joint profiling of histone modifications and transcriptomes. Nat Struct Mol Biol 30, 1428–1433 (2023).\n",
    "62. Yuan, J. & Sims, P. A. An Automated Microwell Platform for Large-Scale Single Cell RNA-Seq. Sci Rep 6, 33883 (2016).\n",
    "63. Zhang, K. et al. A single-cell atlas of chromatin accessibility in the human genome. Cell 184, 5985-6001.e19 (2021).\n",
    "64. Zheng, G. X. Y. et al. Massively parallel digital transcriptional profiling of single cells. Nat Commun 8, 14049 (2017)."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
