{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Protocol for qPCR Validation of ChronoSeq results\n",
    "\n",
    "### Reagents to purchase\n",
    "+ [SingleShot Cell Lysis Kit, 100 x 50 µl rxns #1725080](https://www.bio-rad.com/en-us/sku/1725080-singleshot-cell-lysis-kit-100-x-50-ul-rxns?ID=1725080)\n",
    "+ [iTaq™ Universal SYBR® Green One-Step Kit, 100 x 20 µl rxns, 1 ml #1725150](https://www.bio-rad.com/en-us/sku/1725150-itaq-universal-sybr-green-one-step-kit-100-x-20-ul-rxns-1-ml?ID=1725150)\n",
    "+ [PrimePCR™ SYBR® Green Assay: NFKBIA, Human](https://commerce.bio-rad.com/en-us/prime-pcr-assays/assay/qhsaced0042829-primepcr-sybr-green-assay-nfkbia-human)\n",
    "+ [PrimePCR™ SYBR® Green Assay: ACTB, Human](https://commerce.bio-rad.com/en-us/prime-pcr-assays/assay/qhsaced0036269-primepcr-sybr-green-assay-actb-human)\n",
    "\n",
    "### Instruments used: \n",
    "+ Bio-Rad CFX Opus 96 \n",
    "\n",
    "### Cell Culture and Cell Preparation\n",
    "+ We will need to culture THP-1 cells and K562 Cells for this experiment. \n",
    "+ Make sure you have enough K562 Cells on the day of the experiment to make at least 15ml of K562 at 430cells/&mu;l\n",
    "+ We will follow the [Bowdish lab protocol for culturing THP-1 Cells](http://www.bowdish.ca/lab/wp-content/uploads/2011/07/THP-1-propagation-culture.pdf)\n",
    "+ Make complete growth media with 10%FBS, 1%PEN-STREP, and 0.05mM (1X)[2-Mercaptoethanol](https://www.fishersci.com/shop/products/gibco-2-mercaptoethanol-55mm/21985023) with [ATCC Formulation of RPMI 1640](https://www.fishersci.com/shop/products/gibco-rpmi-1640-medium-atcc-modification-hepes-liquid/A1049101).\n",
    "+ The goal of this experiment is to see if K562 cells can be stimulated by THP-1 Cells through Paracrine signalling.\n",
    "+ We will use vented Tissue Culture Treated T25 Flasks for this experiment.\n",
    "+ Four days before the experiment(prefrerably on a Friday) prepare four T25 flasks with 3ml of THP-1 cells at 430cells/&mu;l.\n",
    "    + Label these flasks 3hr, 2hr, 1 hr and 0hr.\n",
    "+ Make PMA Stock by adding 1&mu;l of 10mM PMA to 1ml of Culture Media. This will give you a 10&mu;M stock. We need to make the final concentration 100nM for the flasks. So for this purpose we will add 30&mu;l of this stock to each Flask.\n",
    "+ Now leave the cells to adhere and differenciate and come back on Monday. On Monday we will:\n",
    "    + Remove the Media and dead cells then wash the cells twice with 3ml of fresh Culture Media.\n",
    "    + Be careful not to detach and cells. Make sure the final volume is still 3ml.\n",
    "    + Leave the cells for 24 hours to rest.\n",
    "+ We will now \n",
    "    \n",
    "### Cell Lysis\n",
    "+ \n",
    "+ Prepare [1000cells/&mu;l of K562 Cells suspended in DMEM](protocol_for_preparing_cells_AdditionalFiltration.ipynb) and load into the ChronoSeq device.\n",
    "+ We will take [Cell samples every 10minutes using the ChronoSeq device](protocol_and_software_for_running_chronoseq_device.ipynb#bulk_cell) for 2 hours.\n",
    "+ We will follow a protocol [similar to SingleShot Cell Lysis Kit](https://www.bio-rad.com/sites/default/files/webroot/web/pdf/lsr/literature/10000112808.pdf) but we will not change our media to DPBS.\n",
    "+ Prepare 12 PCR Tubes for Cell Lysis as follows and keep these Tubes on ice while your prepare. For each tube:\n",
    "    + Label the Tubes 1 through 12 for each timepoint.\n",
    "    + 48&mu;l Single Shot Cell Lysis Buffer\n",
    "    + 1&mu;l Proteinase K Solution\n",
    "    + 1&mu;l DNase Solution\n",
    "+ Start taking samples using the ChronoSeq device. Take 12 samples with a 10minute interval.\n",
    "    + Add 10ng/&mu;l TNF&alpha; to the Cell Suspension after the first sample. Avoid using the same pipettes used for making Lysis solution. \n",
    "    + Adding 5 minutes after Schedule start is ideal.\n",
    "    + Record the time of addition with respect to the first sample. \n",
    "+ For each tube:\n",
    "    + Add 5&mu;l of Cell Suspension. Mix Well. This is equivalent to 5000 Cells.\n",
    "    + Keep the tube back on ice till you have collected all 12 Timepoints.\n",
    "+ Spin down all the Tubes.\n",
    "    + Using a PCR Machine run the following program:\n",
    "        + Lid Temperature 105&deg;C\n",
    "        + Volume 55&mu;l\n",
    "        + Incubate for 10 min at 21&deg;C\n",
    "        + Followed by 5 min at 37°C,\n",
    "        + 5 min at 75°C.\n",
    "        + 4&deg;C for &infin;\n",
    "    + The cell lysate can be stored on ice for up to 4 hr, at -20°C for up to 2 months, or at -80°C for up to 12 months.\n",
    "    + After PCR Immediately store the Tubes at -80&deg;C\n",
    "\n",
    "### Single Step RT-qPCR\n",
    "+ Adapted [from existing protocol here](https://www.bio-rad.com/webroot/web/pdf/lsr/literature/10032048.pdf).\n",
    "+ The iTaq kit comes with ROX as a background reference dye so remember to set that in your qPCR machine. ROX is not necessary for Bio-Rad CFX Opus 96. \n",
    "+ For each gene you will have to setup 12 qPCR reactions. For each 20&mu;l reaction PCR Tube:\n",
    "    + iTaq™ Universal SYBR® Green reaction mix 2X: 10&mu;l\n",
    "    + 1&mu;l PrimePCR Assay primers (20X).\n",
    "    + iScript reverse transcriptase 0.25&mu;l\n",
    "    + 4&mu;l Cell Lysate.\n",
    "    + 4.75&mu;l Nuclease Free Water\n",
    "+ Setup the qPCR reaction as follows: \n",
    "    + 10min at 50&deg;C for Reverse Transcription.\n",
    "    + 1min at 95&deg;C for DNA denaturation and Polymerse activation.\n",
    "    + 40 Cycles of:\n",
    "        + 95&deg;C 10seconds Denaturation\n",
    "        + 60&deg;C 30seconds Annealing and Plate Read\n",
    "+ Here is a screenshot of the plate setup <br><img src=\"qPCR_validation_files/Plate%20Structure%20for%20Validation.PNG\">\n",
    "+ After the plate setup. The plate was sealed with a Optically clear adhesive film.\n",
    "    + Make sure the lips are completely sealed to avoid leaks. Then vortex the plate.\n",
    "+ Plate was then spun down and loaded into the qPCR machine.\n",
    "+ Run the Program.\n",
    "+ Relevant files and results from Bio-Rad CFX Opus 96 can be [found in this folder](qPCR_validation_files).\n",
    "+ Here is an image of the 100/Cq plots from the results. A time delay for ICAM1 and TNFRSF9 can be seen.<br><img src=\"qPCR_validation_files/qPCR_GenePlots_Relative_concentration_100DivCq.svg\">\n",
    "+ You can look at the data analysis [notebook here](qPCR_validation_files/qPCR%20Analysis.ipynb).\n",
    "+ In the Gene Expression data it seems like TNFRSF9 is activated a little later compared to what we saw with the qPCR results. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
