{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Library Preparation Protocol for Experiments with Dropseq beads or ChronoSeq Beads \n",
    "#### Most of this protocol is very similar to [Drop-Seq V3.1](Online-Dropseq-Protocol-v.-3.1-Dec-20152.pdf)\n",
    "### Buffers to Make in Advance\n",
    "Use [Distilled Water](https://www.fishersci.com/shop/products/invitrogen-ultrapure-dnase-rnase-free-distilled-water-2/10977023) for all these Buffers. Please also filter using a [40&mu;m Cell Strainer](https://www.fishersci.com/shop/products/falcon-cell-strainers-4/087711).\n",
    "1. 6X [SSC Buffer](https://www.fishersci.com/shop/products/ambion-ssc-20x-3/AM9763)\n",
    "2. TE-SDS:\n",
    "    + 10 mM [Tris HCl pH 8.0](https://www.fishersci.com/shop/products/1m-tris-hcl-ph-8-0-1l/50843267) + 1 mM [EDTA](https://www.fishersci.com/shop/products/ambion-edta-0-5m-ph-8-0-3/am9261)\n",
    "    + 0.5% [SDS](https://www.fishersci.com/shop/products/sds-solution-20-sodium-dodecyl-sulfate-solution-molecular-biology-electrophoresis-fisher-bioreagents/BP1311200)\n",
    "3. TE-TW:\n",
    "    + 10 mM Tris HCl pH 8.0 + 1 mM EDTA\n",
    "    + 0.01% [Tween-20](https://www.gbiosciences.com/Protein-Research/Tween-20) \n",
    "4. 10 mM Tris HCl pH 8.0\n",
    "5. IDTE (Buffer for Disolving Oligos Ordered from IDT):\n",
    "    + 10 mM Tris HCl pH 8.0 + 0.1mM EDTA\n",
    "\n",
    "### Primers to Order in Advance\n",
    "\n",
    "|**Name**|**Sequence**|**Ordering Parameters for [IDT](https://www.idtdna.com/site/order/oligoentry)**|\n",
    "|-|-|-|\n",
    "|Template Switch Oligo (TSO)|AAGCAGTGGTATCAACGCAGAGTGAATrGrGrG| Order as 100nmole Custom RNA Oligo with HPLC Purification|\n",
    "|SMART PCR primer|AAGCAGTGGTATCAACGCAGAGT| Order as 100nmole Custom DNA Oligo with HPLC Purification|\n",
    "|New-P5‐SMART PCR hybrid oligo|AATGATACGGCGACCACCGAGATCTACACGCCTGTCCGCGGAAGCAGTGGTATCAACGCAGAGT\\*A\\*C| Order as 100nmole Custom DNA Oligo with HPLC Purification| \n",
    "|Custom Read 1 primer|GCCTGTCCGCGGAAGCAGTGGTATCAACGCAGAGTAC| Order as 100nmole Custom DNA Oligo with HPLC Purification| \n",
    "\n",
    "### How to Dissolve Primers/Oligos in IDTE\n",
    "+ Spin down the Tubes you recieved from IDT with your Oligos at the Maximum Possible Speed. There is a small amount of Oligos in the tube and you want them to be pelleted to the bottom of your tubes for easy mixing with IDTE.\n",
    "+ The Spec Sheet for the primers you ordered generally tells you how much IDTE to add to get 100&mu;M concentration Oligos.\n",
    "+ Add the Required volume of IDTE and Pipette up and down many times to dissolve the Oligos and to mix well.\n",
    "    + The Oligos are in a Gel like form and can be stuck to the sides.\n",
    "    + Visually inspect, to make sure you dissolve the Oligos stuck to the sides of the Tubes.\n",
    "+ Make the Correct Dilution and Concentration for the Protocol. In some cases you need 100&mu;M and in others you need 10&mu;M. Made an additional Dilution if you need 10&mu;M.\n",
    "\n",
    "### Best Practices:\n",
    "1. Remember to put back the reagents back to their storage location at 4&deg;C or -20&deg;C once you are done with using them.\n",
    "2. Get fresh new microtips/sterile-disposable pipettes after each step, unless mentioned otherwise. \n",
    "      + Important to avoid cross-contamination.\n",
    "\n",
    "### Equipment Used:\n",
    "+ [Qubit 3.0 Fluorometer](https://tools.thermofisher.com/content/sfs/manuals/qubit_3_fluorometer_man.pdf) (Discontinued Product)\n",
    "+ [PCR Machine](https://www.fishersci.com/shop/products/miniamp-thermal-cycler-rex-extended-warranty-package-2/A37834)\n",
    "+  Our lab uses the refrigerated [Sorval ST8R Centrifuge](https://www.thermofisher.com/order/catalog/product/75007200). You can find [CAD files to print some of the parts here](CAD_Files/Sorval%20ST8R%20CAD%20Models).\n",
    "    + We bought the 50ml Inserts and Printed the Rest.\n",
    "    + You can modify the CAD files to print the 50ml Inserts as well. \n",
    "    + Use PLA or TPU for the prints.\n",
    "+ Microcentrifuge we used was the [Fisherbrand™ accuSpin™ Micro 17/17R Microcentrifuge](https://www.fishersci.com/shop/products/fisherbrand-accuspin-micro-17r-microcentrifuge/13100675)\n",
    "+ [Labnet Rotating Mixer](https://www.stellarscientific.com/labnet-h5600-revolver-with-rotisserie-for-36-x-1-5-2-0ml-tubes-120v/)\n",
    "+ [CELLTREAT Pipette Controller](https://www.fishersci.com/shop/products/pipet-controller-120v-1-cs/50550475)\n",
    "    + [Replacement Filters](https://www.celltreat.com/product/230206/)\n",
    "+ [Labnet Vortex](https://www.stellarscientific.com/labnet-vx-200-s0200-low-profile-lab-vortex-mixer-115v/)\n",
    "+ [Eppendorf Pipettes 6-Pack](https://www.eppendorf.com/us-en/eShop-Products/Liquid-Handling/Manual-Pipettes-Dispensers/Eppendorf-Research-plus-p-3123000942)\n",
    "+ VWR Scientific Products 1545 General Purpose Incubator (Discontinued Product)\n",
    "+ Pipette Tips:\n",
    "    + [Genesee Scientific 200/20&mu;l Tips](https://geneseesci.com/shop-online/product-details/23-150RS/olympus-plastics-23-150rs-200ul-olympus-premium-pipet-tips-low-binding-racked-sterile-10-racks-of-96-tips---unit)\n",
    "    + [Rainin 1000&mu;l Tips](https://www.shoprainin.com/Pipette-Tips/Universal-Pipette-Tips/Hinged-Cover-Racks/Pipette-Tips-RT-UNV-1000%C2%B5L-LS-768A-8/p/30389169)\n",
    "    + [Genesee Scientific 10&mu;l Tips](https://geneseesci.com/shop-online/product-details/23-130RS/olympus-plastics-23-130rs-10-and-#xb5l-olympus-premium-pipet-tips-low-binding-racked-sterile-10-racks-of-96-tips-unit)\n",
    "+ Larger Volume Pipettes:\n",
    "    + [5ml Pipette](https://www.fishersci.com/shop/products/stripette-paper-wrapped-disposable-polystyrene-serological-pipettes-7/07200573)\n",
    "    + [10ml Pipette](https://www.fishersci.com/shop/products/sterile-polystyrene-disposable-serological-pipets-magnifier-stripe-10/1367811F)\n",
    "    + [25ml Pipette](https://www.fishersci.com/shop/products/stripette-paper-wrapped-disposable-polystyrene-serological-pipettes-7/07200575)\n",
    "    + [50ml Pipette](https://www.fishersci.com/shop/products/sterile-polystyrene-disposable-serological-pipets-magnifier-stripe-10/1367811F)\n",
    "+ Disposable Sterile Falcon Tubes:\n",
    "    + [50ml Tubes](https://ecatalog.corning.com/life-sciences/b2c/US/en/Liquid-Handling/Tubes,-Liquid-Handling/Centrifuge-Tubes/Falcon%C2%AE-Conical-Centrifuge-Tubes/p/352098)\n",
    "    + [15ml Tubes](https://ecatalog.corning.com/life-sciences/b2c/US/en/Liquid-Handling/Tubes,-Liquid-Handling/Centrifuge-Tubes/Falcon%C2%AE-Conical-Centrifuge-Tubes/p/352097)\n",
    "+ [Disposable Sterile 250ml GL45 Bottles, Individually Wrapped](https://www.fishersci.com/shop/products/corning-pet-storage-bottles-square-cap-4/07201601)\n",
    "\n",
    "### Preparation for Droplet Breakage and Reverse Transcription \n",
    "+ After the Droplet Collection using the Chrono-Seq Device is complete, close the cap for the Good Collection Tube and put it in the ice bucket. \n",
    "+ Set an incubator to 42&deg;C for the Reverse Transcriptase Reaction.\n",
    "+ The Droplets should look pink in color reflecting the color of the Cell Culture Media.\n",
    "+ Get the [Maxima H Minus Reverse Transcriptase](https://www.fishersci.com/shop/products/thermo-scientific-maxima-h-minus-reverse-transcriptase-3/FEREP0752), [RNAase Inhibitor](https://shop.biosearchtech.com/pcr-instruments%2c-reagents-and-consumables/pcr-and-qpcr-reagents/rnase-inhibitors/nxgen-rnase-inhibitor/p/30281-1), Template Switch Oligo(TSO), [5X RT Buffer](https://www.fishersci.com/shop/products/promega-m-mlv-reverse-transcriptase-7/PRM5313) and [dNTP mix](https://www.fishersci.com/shop/products/deoxynuc-mix-8-umol-of-each/50811740) from -20&deg;C.\n",
    "+ Leave the TSO and dNTP on your bench to melt and keep the RT enzyme and RNAase inhibitor in an ice bucket.\n",
    "+ After the TSO and dNTP have melted, also keep them in the ice bucket.\n",
    "+ After the RT buffer has melted, make 2X RT buffer by mixing 1 ml of 5X RT buffer with 1.5ml of Distilled Water.\n",
    "+ Put the 2X RT buffer in the ice bucket.\n",
    "+ You will also need [Rainin Pipette Tips](https://www.shoprainin.com/Pipette-Tips/Universal-Pipette-Tips/Hinged-Cover-Racks/Pipette-Tips-RT-UNV-1000%C2%B5L-LS-768A-8/p/30389169) for the P1000.\n",
    "\n",
    "\n",
    "### Droplet Breakage\n",
    "\n",
    ">Execute the Cell below to see videos of the Droplet Breakage Process. Please read the protocol below for more details."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table><tr><th>Remove maximum oil Possible without taking up any oil</th>\n",
       "<th>Add 30ml Room Temperature 6X SSC Buffer<th></tr>\n",
       "<tr><td><iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/gQ23bTDwGoY?si=LMYQ2Riyin3KO2dv\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td>\n",
       "<td><iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/UUhx5wkZd9g?si=itTsmt1kOniAynEZ\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td></tr></table>\n",
       "<table><tr><th>Add 1ml PFO</th>\n",
       "<th>Forceful Vertical Shakes to Break Droplets<th></tr>\n",
       "<tr><td><iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/aFuPUB2Rhzc?si=oJ0bQouwdDVKpgZ-\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td><td>\n",
       "<iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/h9I364NjFIQ?si=w9Rz_59j-yc-Kwp2\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td></tr></table>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%HTML\n",
    "<table><tr><th>Remove maximum oil Possible without taking up any oil</th>\n",
    "<th>Add 30ml Room Temperature 6X SSC Buffer<th></tr>\n",
    "<tr><td><iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/gQ23bTDwGoY?si=LMYQ2Riyin3KO2dv\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td>\n",
    "<td><iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/UUhx5wkZd9g?si=itTsmt1kOniAynEZ\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td></tr></table>\n",
    "<table><tr><th>Add 1ml PFO</th>\n",
    "<th>Forceful Vertical Shakes to Break Droplets<th></tr>\n",
    "<tr><td><iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/aFuPUB2Rhzc?si=oJ0bQouwdDVKpgZ-\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td><td>\n",
    "<iframe width=\"450\" height=\"250\" src=\"https://www.youtube.com/embed/h9I364NjFIQ?si=w9Rz_59j-yc-Kwp2\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\" allowfullscreen></iframe></td></tr></table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Remove the oil layer from the bottom of the Good Collection Tube by pressing a P1000 down to its first stop, pushing through the droplets to the bottom of the tube, pressing down to the second stop to expel any droplets, and then after waiting several seconds for the droplets to float back up to the droplet layer, sucking out the oil. You do not need to remove every last bit of oil just remove most of it.\n",
    "+ Add 30 mL of room temperature 6X [SSC buffer](https://www.fishersci.com/shop/products/ambion-ssc-20x-3/AM9763).\n",
    "+ Add 1 mL of [Perfluorooctanol (PFO)](https://www.fishersci.com/shop/products/1h-1h-2h-2h-perfluorooctanol-97-thermo-scientific/AAB2015618)(**Warning: Hazardous Chemical, read SDS**).  Shake by hand to break the droplets (3‐4 forceful vertical shakes).\n",
    "+ Slightly loosen the cap and put the tube in a centrifuge with Soft Deceleration. The Centrifuge should be set to 1000xg for 1:30 minutes with Soft Deceleration Enabled. If possible store this as a program.\n",
    "+ Carefully Remove the tube from the Centrifuge. Without disturbing the beads on the interface, use a 25ml pipette to remove and discard the supernatant on top until there are only a few mL remaining above the interface.\n",
    "+ Label a **new Falcon Tube** with the Name of the Experiment.\n",
    "+ Tilt the falcon tube to a 30 degree angle from the vertical.\n",
    "    + Add 30 mL of 6X SSC to the wall, to kick up the beads into solution.\n",
    "    + If you do this correctly almost no oil and only beads will be kicked into solution.\n",
    "    + Wait a few seconds to allow the majority of the oil to sink to the bottom, then transfer the supernatant to the **new Falcon tube**.\n",
    "    + Avoid transferring any oil or interface precipitate material. You should be able to see the white beads floating around in the supernatant during this step.\n",
    "+ Spin down the beads in the new Falcon Tube. The Centrifuge should be set to 1000xg for 1:30 minutes with Soft Deceleration Enabled.\n",
    "+ Discard the Contents of the Old 50ml Falcon Tube with the remaining Oil, PFO and 6X SSC buffer into the [Waste Bottle under your Bench](protocol_and_software_for_running_chronoseq_device.ipynb#Discard-into-Waste-Bottle).\n",
    "+ Label a new [1.5ml eppendorf tube](https://www.fishersci.com/shop/products/eppendorf-snap-cap-microcentrifuge-biopur-safe-lock-tubes-3/0540224B) (eppie) with the name and date of the experiment.\n",
    "+ The beads are now pelleted to the very bottom of the new 50ml Falcon tube. **Without distrubing the beads** carefully remove all but ~1 mL of liquid.\n",
    "+ Without touching the beads, slightly pipette up and down using the P1000 to resuspend the beads in solution. Using the same pipette tip transfer the liquid to the labeled eppie.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ If you see any oil left in the eppie you need to remove it. The beads will appear slightly higher than the base of the Tube if there is oil left: \n",
    "    + You will first need a new clean eppie.\n",
    "    + Pipette up and down slightly with the P1000 to first resuspend the beads in solution. Then quickly and carefully remove almost all the liquid except the oil.\n",
    "    + Transfer this liquid to the second clean eppie.\n",
    "\n",
    "### Reverse Transcription\n",
    "+ Carefully remove all but ~100&mu;l of Liquid from the eppie, without disturbing the beads.\n",
    "+ Wash the beads twice with 2X RT buffer:\n",
    "    + Add 1ml of 2X RT buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of 2X RT buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"bulk_continuation\"></a>\n",
    "\n",
    "> Optional Note: [Bulk Assay mentioned in a separate notebook](protocol_for_bulk_rna_seq_library_prep.ipynb) , continues from this point onwards."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "+ Remove 500&mu;l of Liquid from the eppie without disturbing the beads.\n",
    "+ Without touching the beads, slightly pipette up and down using the P1000 to resuspend the beads in solution.\n",
    "+ Now estimate the Volume in the eppie by using the P1000.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Depending on the Type of Experiment you will need to split your beads into smaller reactions:\n",
    "    + For 12 Timepoint Single-Cell Experiments split the beads into 4 Equal Parts for Reverse Transcription.\n",
    "    + For 6 Timepoint Single-Cell Experiments split the beads into 2 Equal Parts for Reverse Transcription.\n",
    "    + For 3 Timepoint Single-Cell Experiments just a Single Reaction should be sufficient.\n",
    "    + For all Bulk Experiments including 12 Timepoint Bulk Experiments a Single Reaction is sufficient.\n",
    "\n",
    "> For reasons we don't full understand yet. Different Reverse Transcription reactions from the same experiment had different qubit concentrations and unique transcripts captured. However, this batch effect was not seen for PCR Splits from the same RT Reaction. \n",
    "\n",
    "+ Using the estimated volume in the eppie, make the final volume of the eppie 100&mu;l without disturbing the beads.\n",
    "+ Then add the following to the eppie. **Pipette up and down to mix well after adding each component**:\n",
    "\t- 40&mu;l of 10mM dNTPs\n",
    "\t- 20&mu;l of 100um Template Switch Oligo (TSO)\n",
    "\t- 5&mu;l RNAase inhibitor\n",
    "\t- 10&mu;l Maxima H Minus Reverse Transcriptase\n",
    "    - 25&mu;l H<sub>2</sub>O\n",
    "+ Vortex the beads for 3 seconds at the maximum setting.\n",
    "+ Incubate at Room temperature for 30 minutes with Rotation.\n",
    "+ Incubate at 42&deg;C for 90 minutes with Rotation. (TBD: 30min Room Temp, 42C for 45min, 30min at 50C then 15min at 65C then 15 min at Room temperature)\n",
    "> &sext; **Note**: It is a good idea to finish the Cleanup for the ChronoSeq Device for [Single Cell](protocol_and_software_for_running_chronoseq_device.ipynb#cleanup_singlecell) or [Bulk experiments](protocol_and_software_for_running_chronoseq_device.ipynb#cleanup_bulk) while you are waiting for the Reverse Transcription Reaction to Complete.\n",
    "\n",
    "+ Wash the beads once with TE-SDS and then twice with TE-TW:\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Add 1 ml of TE-SDS to the eppie.\n",
    "    + Flip the tube upside down a few times and then Vortex the beads at full speed for 2 seconds. \n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of TE-TW. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of TE-TW. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "> This is a **STOPPING STEP**. You can store the Beads in TE-TW at 4&deg;C and resume the protocol later if you want.\n",
    "\n",
    "\n",
    "### Note on PCR Splits and Noise\n",
    "1. For 12 Time-Tag Bulk Experiments we recommend two splits.\n",
    "2. For 3 Time-Tag Single Cell Experiments we recommend no splits.\n",
    "3. for 12 Time-Tag Single Cell Production runs we recommend 4 Splits (with each RT reaction being its own split RT1->Split1 etc.).\n",
    "\n",
    "### Suggested PCR Cycles for the first round of PCR\n",
    "Anything between 11PCR cycles to 14 PCR Cyles works well. Larger cells with more RNA genererally need fewer PCR Cycles. \n",
    "As a baseline we recommend 14 Cycles. \n",
    "\n",
    "### Strategies for Increasing qubit DNA Conentration\n",
    "+ You can try to take out as much liquid as possible from the PCR Tubes for the first 0.6X Ampure purification below.\n",
    "+ Elute in 10&mu;l of Distilled Waster for last step to increase concentration further. We need a final concentration of preparaby 0.120ng/&mu;l or higher.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PCR Preparation\n",
    "+ Get the [KAPA HiFi HotStart ReadyMix](https://www.fishersci.com/shop/products/hifi-hotstart-ready-mix-100rxn-1/501965217) and SMART PCR Primer from -20&deg;C and leave them on your bench to melt. After they have melted put them in an ice bucket.\n",
    "+ Decide the number of splits needed for PCR and get the same number of 200&mu;l PCR tubes ready. Label each PCR tube with the split number. \n",
    "    + In general just putting all the beads for an experiment into a single PCR tube/split works quite well for us.\n",
    "+ Wash the beads twice with Distilled Water:\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1 ml of Distilled Water to the eppie. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1 ml of Distilled Water to the eppie. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "+ Without touching the beads, slightly pipette up and down using the P200 to resuspend the beads in solution.\n",
    "+ Now estimate the Volume in the eppie by using the P200.\n",
    "+ *(Optional)* If you want to do Splits:\n",
    "    + Transfer the estimated volume of beads to one of the PCR tubes. \n",
    "    + Evenly resuspend the beads using the P200 and then transfer equal volumes to each labeled PCR tube.\n",
    "+ Spin all the PCR tubes down using a Centrifuge with Soft decceleration capability. The Centrifuge should be set to 1000xg for 1:30 minutes with Soft Deceleration Enabled.\n",
    "+ Make the Final Volume of each PCR tube 24.6&mu;l by removing liquid without disturbing the beads.\n",
    "+ Each PCR tube should have the following:\n",
    "    + 24.6&mu;l beads in Distilled Water\n",
    "    + 0.4&mu;l 100&mu;M SMART PCR Primer\n",
    "    + 25&mu;l 2X KAPA HiFi HotStart ReadyMix\n",
    "+ Pipette up and down to mix well.\n",
    "+ Make sure there is no water condensation inside your PCR wells for your PCR machine.\n",
    "> Optional: You can set your well temperature to 98&deg;C for 5 minutes with the lid open to get rid of the condensation.\n",
    "+ Put the PCR Tubes inside your PCR Machine."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id=\"dropseq_pcr\"></a>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### PCR Program\n",
    "Now run the Following PCR Program:\n",
    "> Lid Temperature 105&deg;C\n",
    "><br> Volume of Liquid: 50&mu;l\n",
    "\n",
    "<br>95&deg;C 3 minutes\n",
    "\n",
    "\n",
    "##### 4 cycles of:\n",
    "   + 98&deg;C 20 s\n",
    "   + 65&deg;C 45 s\n",
    "   + 72&deg;C 3 min\n",
    "   \n",
    "##### 10 cycles of: \n",
    "   + 98&deg;C 20 s\n",
    "   + 67&deg;C 20 s\n",
    "   + 72&deg;C 3 min\n",
    "\n",
    "<br> \n",
    "72&deg;C 5 min <br>\n",
    "4&deg;C &infin;\n",
    "\n",
    "> Store at -20&deg;C if you cannot complete the Cleanup the same day. **From experience we have seen libraries degrade if left at 4&deg;C without cleanup. Its very important to finish the PCR Cleanup as soon as possible.** Do not delay the cleanup.\n",
    "\n",
    "### PCR Cleanup\n",
    "+ Get the [AMpure XP Beads](https://www.beckman.com/reagents/genomic/cleanup-and-size-selection/pcr/a63880) from 4&deg;C.\n",
    "+ Vortex the AMpure beads at full speed for 10 seconds to resuspend the beads completely. \n",
    "+ If you haven't already done this: Take 0.5ml of the AMpure XP beads and transfer them to a 1.5ml Eppendorf Tube. Label the tube with the expiration date and the contents.\n",
    "+ Put the main AMpure XP beads bottle back to 4&deg;C.\n",
    "+ Leave the Eppendorf Tube with the AMpure XP beads aliquot on your bench till it reaches room temperature.\n",
    "+ Spin all the PCR tubes down using a Centrifuge with Soft decceleration capability. The Centrifuge should be set to 1000xg for 1:30 minutes with Soft Deceleration Enabled.\n",
    "> &#128161; **Tip:** Our lab uses the refrigerated [Sorval ST8R Centrifuge](https://www.thermofisher.com/order/catalog/product/75007200). You can find [CAD files to print some of the parts here](CAD_Files/Sorval%20ST8R%20CAD%20Models).\n",
    "+ Prepare another set of clean PCR tubes identical in number to the PCR tubes used for running the PCR reaction. \n",
    "+ Transfer 40&mu;l of Liquid from each PCR tube to its corresponding new PCR tube, without disturbing the beads.\n",
    "+ Prepare 25ml of fresh 80% Ethanol by mixing 20ml of 100% Ethanol with 5ml of Distilled Water in a 50ml Falcon Tube. Label the tube with the date and contents.\n",
    "+ We will do two rounds of purifications. First will be a 0.6X AMpure Fraction and the second will be a 0.9X AMpure Fraction.\n",
    "\n",
    "#### First round 0.6X AMpure Fraction:\n",
    "+ Vortex the eppie with the AMpure XP beads to evenly resuspend the beads.\n",
    "+ Add 24&mu;l of AMpure XP beads to each 40&mu;l PCR tube. This is a 0.6X AMpure beads fraction.\n",
    "+ Mix each PCR tube well by pipetting up and down at least 15 times.\n",
    "+ Incubate at room temperature for 5 minutes.\n",
    "+ Place the PCR tubes in a [Magnetic Rack](https://www.amazon.com/gp/product/B081373C58). Wait two minutes for the liquid to become clear.\n",
    "+ Don't remove the PCR tubes from the Magnetic Rack.\n",
    "+ Remove 62&mu;l of liquid from each PCR tube, without disturbing the AMpure XP beads.\n",
    "+ Wash the beads with 80% Ethanol twice:\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the AMpure XP beads.\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the AMpure XP beads.\n",
    "+ Using a P200, remove the residual liquid from each PCR tube without disturbing the AMpure XP beads.\n",
    "+ Leave the PCR tubes on the Magnetic rack for 10-15 minutes to allow the remaining Ethanol to evaporate.\n",
    "+ Remove the PCR tubes from the Magnetic rack.\n",
    "+ Now add 52&mu;l of Distilled Water to each PCR tube. Using the same tip evenly resuspend all the beads stuck to the side of the PCR tube.\n",
    "+ Incubate at room temperature for 2 minutes.\n",
    "+ Place all the PCR tubes back in the Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "\n",
    "#### Second round 0.9X AMpure Fraction:\n",
    "+ Prepare another set of clean PCR tubes identical in number to the PCR tubes in the Magnetic rack.\n",
    "+ Without sucking up any AMpure beads, transfer 50&mu;l of liquid from each PCR tube in the rack to its corresponding clean PCR tube.\n",
    "+ Vortex the eppie with the AMpure XP beads to evenly resuspend the beads.\n",
    "+ Add 45&mu;l of AMpure XP beads to each 50&mu;l PCR tube. This is a 0.9X AMpure beads fraction.\n",
    "+ Mix each PCR tube well by pipetting up and down at least 15 times.\n",
    "+ Incubate at room temperature for 5 minutes.\n",
    "+ Place the PCR tubes in a Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "+ Don't remove the PCR tubes from the Magnetic Rack.\n",
    "+ Remove 93&mu;l of liquid from each PCR tube, without disturbing the AMpure XP beads.\n",
    "+ Wash the beads with 80% Ethanol twice:\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the AMpure XP beads.\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the AMpure XP beads.\n",
    "+ Using a P200, remove the residual liquid from each PCR tube without disturbing the AMpure XP beads.\n",
    "+ Leave the PCR tubes on the Magnetic rack for 10-15 minutes to allow the remaining Ethanol to evaporate.\n",
    "+ Remove the PCR tubes from the Magnetic rack.\n",
    "+ Now add 13&mu;l of Distilled Water to each PCR tube. Using the same tip evenly resuspend all the beads stuck to the side of the PCR tube.\n",
    "+ Incubate at room temperature for 2 minutes.\n",
    "+ Place all the PCR tubes back in the Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "+ Prepare clean 1.5ml Eppendorf tubes identical in number to the PCR tubes in the Magnetic rack. Label each tube with PCR split number, a short description of the experiment, your name and the date.\n",
    "+ Without sucking up any AMpure Xp beads, transfer 11&mu;l of liquid from each PCR tube in the rack to its corresponding eppie.\n",
    "\n",
    "#### cDNA Library Concentration with Qubit\n",
    "\n",
    "> &#x26A0; **IMPORTANT**:For all Qubit Tubes **only Pipette till the first stop** even if there is liquid left in the pipette tip. **Don't pipette up and down to mix.**\n",
    "\n",
    "+ Get the [Qubit 1X dsDNA HS Working Solution](https://www.fishersci.com/shop/products/qubit-1x-dsdna-hs-br-assay-kits/Q33231) from 4&deg;C.\n",
    "> &sext; **Important Note**: Calibrate the Qubit device using the Standards if it hasn't been done recently:\n",
    ">> + Add 190&mu;l each of Qubit 1X dsDNA HS Working Solution to two Qubit Assay Tubes.\n",
    ">> + Add 10&mu;l of Standard 1 to the first Qubit Assay Tube.\n",
    ">> + Add 10&mu;l of Standard 2 to the second Qubit Assay Tube.\n",
    ">> + Vortex both tubes at full speed for 5 seconds.\n",
    ">> + Wait 2 minutes.\n",
    ">> + Go to the Device Home Screen.\n",
    ">> + Select the 1X dsDNA High Sensitivity assay\n",
    ">> + Select the Option to calibrate the Qubit.\n",
    ">> + Insert the Assay tube with Standard 1 when prompted by the device.\n",
    ">> + Insert the Assay tube with Standard 2 when prompted by the device. \n",
    "+ Prepare [Qubit Assay tubes](https://www.fishersci.com/shop/products/invitrogen-qubit-assay-tubes/Q32856) identical in number to the eppies.\n",
    "+ Add 199&mu;l of Qubit 1X dsDNA HS Working Solution to each Qubit Assay tube.\n",
    "+ Transfer 1&mu;l of purified product from each eppie to its corresponding Qubit Assay Tube.\n",
    "+ Vortex all the Qubit Assay tubes at full speed for 5 seconds.\n",
    "+ Wait 2 minutes.\n",
    "+ Go to the Device Home Screen. Select the 1X dsDNA High Sensitivity assay\n",
    "+ Select 1&mu;l as Sample Volume.\n",
    "+ Now measure the dsDNA concentration for each tube in ng/&mu;l using the Qubit device.\n",
    "+ You should expect a concentration between 0.3ng/&mu;l to 1ng/&mu;l for each split if you ran 12 PCR cycles.\n",
    "\n",
    "> This is a **STOPPING STEP**. You can store your eppies with the libraries at -20&deg;C. I recommend having a decicated Storage box for storing these libraries.\n",
    "\n",
    "### Sample Submission for Tapestation HS D5000\n",
    "> These instructions are for researchers working at University of California, San Diego. Please follow the instructions for the Core facilities at your local institution.\n",
    "\n",
    "+ Prepare clean 1.5ml Eppendorf tubes identical in number to the eppies for each split/sample.\n",
    "+ Download and complete the [Tapestation Sample Submission Manifest](https://igm.ucsd.edu/genomics/submission#:~:text=TAPESTATION%20MANIFEST) for Tapestation HS D5000 screentape assay.\n",
    "+ Please follow the instructions in the manifest for labelling your eppies.\n",
    "+ You will need to submit at least 3&mu;l of each split/sample for Tapestation HS D5000.\n",
    "+ Follow the instructions on the [IGM website](https://igm.ucsd.edu/genomics/submission) on how and when to submit your tubes."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Note on Pooling and Indices for the Sequencing on the Same Lane: \n",
    "+ You can choose to pool your PCR splits into a Single Tagmentation Reaction and use the same i7 index for all the splits.\n",
    "    + You can also Tagment them separately with different i7 Indices. This choice is up to you.\n",
    "    + When you combine different splits together then equal amounts of DNA should be added to the Tagmentation reaction from each split. So for example, if you have two splits then 0.30ng should come from each split.\n",
    "+ Different experimental conditions should always have different i7 indices if you are sequencing them on the same lane.\n",
    "+ Always use different indices for Bulk Experiments being sequenced on the same Lane.\n",
    "+ For the [Nextera Index Kit (FC-131-1001)](https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/nextera-xt-dna.html) the i7 Indices are N701-N706. [Other kits can provide even more indices](https://support.illumina.com/sequencing/sequencing_kits/nextera-xt-index-kit-v2/contents-storage.html) if you want more libraries on the Same Lane.\n",
    "+ When you submit your Libraries always use the i7 index Sequence in the \"i7 Bases for Sample Sheet\" Column [mentioned here](https://support-docs.illumina.com/SHARE/AdapterSequences/Content/SHARE/AdapterSeq/Nextera/NexteraDNAIndexes.htm).\n",
    "+ If you have low plex (less than six libraries being pooled) make sure you use the correct combination of i7 Indices for Color compatibility. [Please see Table 2 in this Low Plex Pooling Guide from Illumina](technote-nextera-rapid-capture-low-plex-pooling-guidelines.pdf).\n",
    "\n",
    "### Tagmentation\n",
    "\n",
    "#### Preparing for Tagmentation\n",
    "+ Make sure there is no water condensation inside your PCR wells for your PCR machine.\n",
    "> **OPTIONAL**: You can set your well temperature to 98&deg;C for 5 minutes with the lid open to get rid of the condensation.\n",
    "+ Get the Tagment DNA Buffer and Amplicon Tagment Mix from the [Nextera XT Kit (FC-131-1024)](https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/nextera-xt-dna.html) stored at -20&deg;C\n",
    "+ Leave the Tagment DNA Buffer to melt on your bench. Put the Amplicon Tagment Mix(ATM) in an ice bucket.\n",
    "+ Get the Neutralization(NT) Buffer and also leave it on your bench for later use.\n",
    "+ Preheat a PCR Machine to 55&deg;C:\n",
    "    + Set the Volume to 20&mu;l\n",
    "    + Lid temperature of 105&deg;C\n",
    "    + Duration &infin;.\n",
    "    + I recommend using the Incubate function if your PCR Machine has this option.\n",
    "+ For each sample, prepare 200&mu;l PCR Tubes with 0.6ng of cDNA in a total volume of 5&mu;l:\n",
    "    + Use Qubit Concentration Values when making this dilution.\n",
    "    + Try to avoid Pipetting Volumes less than 0.4&mu;l as the Accuracy is not very good.\n",
    "    + Use the P2.5 Pipette whenever possible for greater accuracy.\n",
    "    + Use Distilled Water for making the dilution.\n",
    "+ You can still go ahead with this reaction even if your DNA concentration is less than 0.120ng/&mu;l. Just add 8&mu;l of whatever you have.\n",
    "\n",
    "\n",
    "#### Run Tagmentation Reaction\n",
    "+ Keep a record of which i7 index (N70X Oligo) will be used with each PCR Tube. \n",
    "+ To each tube, add 10 &mu;l of Tagment DNA buffer.\n",
    "+ Now add 5 &mu;l of Amplicon Tagment Mix to each tube for a total volume of 20&mu;l.\n",
    "+ Mix well by pipetting 5 times.\n",
    "+ Spin down the PCR Tubes using a Tabletop centrifuge for PCR Tubes.\n",
    "+ Incubate @55&deg;C. **Set a timer for 5 minutes.**\n",
    "+ Without delay, Add 5&mu;l of Neutralization Buffer to each tube.\n",
    "+ Mix well by pipetting 5 times. \n",
    "+ Spin down the PCR Tubes using a Tabletop centrifuge for PCR Tubes.\n",
    "+ Incubate at room temperature for 5 minutes.\n",
    "\n",
    "> **NOTE:** If you don't have enough Nextra PCR Mix (NPM) left in your Nextera XT Kit you can use [this alternative protocol](protocol_for_tagmentation_with_KAPA_PCR.ipynb#Preparation-for-PCR) we have already tested using KAPA HiFi Hotstart Ready Mix. Typically Illumina Provides an excess amount of ATM but not enough NPM in their Nextera XT Kit. You can save money by not buying a whole new Nextera Kit when you run out of NPM. \n",
    "\n",
    "#### Preparation for PCR \n",
    "+ Get the Nextera PCR Mix(NPM) from the Nextera XT Kit stored at -20&deg;C. Put it in an ice bucket.\n",
    "+ Get the [Nextera Index Kit (FC-131-1001)](https://www.illumina.com/products/by-type/sequencing-kits/library-prep-kits/nextera-xt-dna.html) from -20&deg;C. Leave the i7 Indices you plan to use for your samples on the bench to melt.\n",
    "> &sext; **Important Note**: Make sure you use the correct combination of i7 Indices for Color compatibility. [Please see Table 2 in this Low Plex Pooling Guide from Illumina](technote-nextera-rapid-capture-low-plex-pooling-guidelines.pdf).\n",
    "+ Get the **10&mu;M** New‐P5‐SMART PCR hybrid oligo from -20&deg;C and leave it on your bench to melt.\n",
    "+ Now add each of these to your PCR tubes in the following order:\n",
    "    1. 5&mu;l **10&mu;M** New‐P5‐SMART PCR hybrid oligo\n",
    "    2. 5&mu;l of the i7 index (N70X Oligo) from the Nextera Index Kit.\n",
    "    3. 15&mu;l of the Nextera PCR Mix (NPM).\n",
    "+ Mix well by pipetting 5 times.\n",
    "+ Spin down the PCR Tubes using a Tabletop centrifuge for PCR Tubes.\n",
    "\n",
    "### Post Tagmentation PCR Program\n",
    "Now run the Following PCR Program:\n",
    "\n",
    "> Lid Temperature 105&deg;C\n",
    "><br> Volume of Liquid: 50&mu;l\n",
    "\n",
    "<br>72&deg;C 3 min\n",
    "<br>95&deg;C 30 s\n",
    "\n",
    "##### 16 cycles of:\n",
    "   + 95&deg;C 10 s\n",
    "   + 55&deg;C 30 s\n",
    "   + 72&deg;C 30 s\n",
    "\n",
    "<br> \n",
    "72&deg;C 5 min <br>\n",
    "4&deg;C &infin;\n",
    "\n",
    "> Store at -20&deg;C if you cannot complete the Cleanup the same day. **From experience we have seen libraries degrade if left at 4&deg;C without cleanup. Its very important to finish the PCR Cleanup as soon as possible.** Do not delay the cleanup.\n",
    "\n",
    "### PCR Cleanup\n",
    "+ Get the Ampure XP Beads eppie from 4&deg;C.\n",
    "+ Vortex the AMpure beads at full speed for 10 seconds to resuspend the beads completely. Leave them on your bench till they reach room temperature.\n",
    "+ Spin down the PCR Tubes using a Tabletop centrifuge for PCR Tubes.\n",
    "+ We will do two rounds of purifications. First will be a 0.6X AMpure Fraction and the second will be a 0.9X AMpure Fraction.\n",
    "\n",
    "#### First round 0.6X AMpure Fraction:\n",
    "+ Vortex the eppie with the Ampure XP beads to evenly resuspend the beads.\n",
    "+ Add 30&mu;l of Ampure XP beads to each 50&mu;l PCR tube. This is a 0.6X AMpure beads fraction.\n",
    "+ Mix each PCR tube well by pipetting up and down at least 15 times.\n",
    "+ Incubate at room temperature for 5 minutes.\n",
    "+ Place the PCR tubes in a Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "+ Don't remove the PCR tubes from the Magnetic Rack.\n",
    "+ Remove 75&mu;l of liquid from each PCR tube, without disturbing the Ampure XP beads.\n",
    "+ Wash the beads with 80% Ethanol twice:\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the Ampure XP beads.\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the Ampure XP beads.\n",
    "+ Using a P200, remove the residual liquid from each PCR tube without disturbing the Ampure XP beads.\n",
    "+ Leave the PCR tubes on the Magnetic rack for 10-15 minutes to allow the remaining Ethanol to evaporate.\n",
    "+ Remove the PCR tubes from the Magnetic rack.\n",
    "+ Now add 52&mu;l of Distilled Water to each PCR tube. Using the same tip evenly resuspend all the beads stuck to the side of the PCR tube.\n",
    "+ Incubate at room temperature for 2 minutes.\n",
    "+ Place all the PCR tubes back in the Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "\n",
    "#### Second round 0.9X AMpure Fraction:\n",
    "+ Prepare another set of clean PCR tubes identical in number to the PCR tubes in the Magnetic rack.\n",
    "+ Without sucking up any AMpure beads, transfer 50&mu;l of liquid from each PCR tube in the rack to its corresponding clean PCR tube.\n",
    "+ Vortex the eppie with the Ampure XP beads to evenly resuspend the beads.\n",
    "+ Add 45&mu;l of Ampure XP beads to each 50&mu;l PCR tube. This is a 0.9X AMpure beads fraction.\n",
    "+ Mix each PCR tube well by pipetting up and down at least 15 times.\n",
    "+ Incubate at room temperature for 5 minutes.\n",
    "+ Place the PCR tubes in a Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "+ Don't remove the PCR tubes from the Magnetic Rack.\n",
    "+ Remove 93&mu;l of liquid from each PCR tube, without disturbing the Ampure XP beads.\n",
    "+ Wash the beads with 80% Ethanol twice:\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the Ampure XP beads.\n",
    "    + Add 200&mu;l of the freshly prepared 80% Ethanol to each tube.\n",
    "    + Wait 30 seconds.\n",
    "    + Remove the liquid from each PCR tube without disturbing the Ampure XP beads.\n",
    "+ Using a P200, remove the residual liquid from each PCR tube without disturbing the Ampure XP beads.\n",
    "+ Leave the PCR tubes on the Magnetic rack for 10-15 minutes to allow the remaining Ethanol to evaporate.\n",
    "+ Remove the PCR tubes from the Magnetic rack.\n",
    "+ Now add 17&mu;l of Distilled Water to each PCR tube. Using the same tip evenly resuspend all the beads stuck to the side of the PCR tube.\n",
    "+ Incubate at room temperature for 2 minutes.\n",
    "+ Place all the PCR tubes back in the Magnetic Rack. Wait two minutes for the liquid to become clear.\n",
    "+ Prepare clean 1.5ml Eppendorf tubes identical in number to the PCR tubes in the Magnetic rack. Label each tube with the i7(N70X) Oligo used, a short description of the experiment, your name and the date.\n",
    "+ Without sucking up any Ampure XP beads, transfer 15&mu;l of liquid from each PCR tube in the rack to its corresponding eppie.\n",
    "\n",
    "#### Tagmented Library Concentration with Qubit\n",
    "> &#x26A0; **IMPORTANT**:For all Qubit Tubes **only Pipette till the first stop** even if there is liquid left in the pipette tip. **Don't pipette up and down to mix.**\n",
    "\n",
    "+ Get the Qubit 1X dsDNA HS Working Solution from 4&deg;C.\n",
    "+ Prepare Qubit Assay tubes identical in number to the eppies.\n",
    "+ Add 199&mu;l of Qubit 1X dsDNA HS Working Solution to each Qubit Assay tube.\n",
    "+ Transfer 1&mu;l of purified product from each eppie to its corresponding Qubit Assay Tube.\n",
    "+ Vortex all the Qubit Assay tubes at full speed for 5 seconds.\n",
    "+ Wait 2 minutes.\n",
    "+ Go to the Device Home Screen. Select the 1X dsDNA High Sensitivity assay\n",
    "+ Select 1&mu;l as Sample Volume.\n",
    "+ Now measure the dsDNA concentration for each tube in ng/&mu;l using the Qubit device.\n",
    "+ You should expect a concentration between 5ng/&mu;l to 18ng/&mu;l for each sample/split.\n",
    "\n",
    "> This is a **STOPPING STEP**. You can store your eppies with the libraries at -20&deg;C. I recommend having a decicated Storage box for storing these libraries.\n",
    "\n",
    "## Option 1: Submitting your Libraries to Admera Health for sequencing:\n",
    "> We found Admera health's service to be excellent and reasonably priced. You can email them at custom-services@admerahealth.com for a quote. However, at the time of writing this they had not discontinued their HiSeq sequencers. They planned to phase out this sequencer soon because Illumina planned to stop supporting it. They accepted Custom Read 1 Primers for a single lane only when using the HiSeq. They wanted us to purchase the whole Flow Cell if we wanted to use the Custom Read 1 Primer for other Illumina sequencers. This is partly because of the removal of cBot by Illumina in NovaSeq. If this remains the case, Admera health might not remain reasonably priced in the future when using a Custom Read 1 Primer. We also contacted our local core facility and there were willing to mix our primer with the Illumina primer. They allowed this even if we didn't purchase a whole flow cell for NovaSeq. This primer mix did not interfere with the sequencing of our sample and the sequencing for other Customers using Illumina primers in other lanes on the same flow cell. The steps below were for sequencing at University of California, San Diego but should be very similar for your local core facility. If your local core facility tells you they are afraid to mix your Custom Read 1 Primer with the Illumina primer, then just tell them we already did this without any problems.\n",
    "\n",
    "\n",
    "<em>Note: Instructions are for Researchers working at UC San Diego. Please adapt these instructions for your institution.</em>\n",
    "+ We recommend submitting your samples to Admera Health for Sequencing. Please request a Quote for a single lane of PE150 using HiSeq. [Here is an example email](UCSD%20Jacobs%20School%20of%20Engineering%20Mail%20-%20Quote%20for%20Sequencing%20and%20Pooling.pdf). Admera is suitable for Quality Control and other test libraries as they also have a quick turnaround time. Use IGM when large throughput and read depth is needed at a reasonable cost.\n",
    "+ **They will pool and run the Tapestation D1000 for your samples** on request at a reasonable price. Please ask them to include this cost in the quote request. \n",
    "+ You will also have to submit:\n",
    "    + 50&mu;l of your 100&mu;M Custom Read 1 Primer in an eppie.\n",
    "    + At least 10&mu;l of each library in a separate eppie. Libraries should be more than 10nM. [Use the Calculator Below](#nm_calculator) to estimate this concentration. Label each eppie on the top with its i7 N70X index.\n",
    "    + Fill the Sample Submission Form and Excel Sheet with Sample Information for Admera Health. [Here is an Example Sheet](Seq_Only_Sample_Info_Template_July2023.xlsx). [Here is an example submission form](UCSD%20Jacobs%20School%20of%20Engineering%20Mail%20-%20We%20have%20received%20your%20response%20for%20Sample%20Submission%20Form%20-%20Sequencing.pdf)\n",
    "    + Print a copy of the Quote and the completed Excel Sheet with Sample Inforamation. Put them in a **small** Ziplock bag. \n",
    "+ Store all these items in a [Cryogenic Storage Box](https://us.vwr.com/store/product/4614621/null).    \n",
    "+ You should ship this storage box in a larger **thermally insulated box with an ice pack**. Please use overnight next day shipping. You can create a new shipping request at logistics.ucsd.edu (VPN required for Off-Campus use) to ship this package. Click on: Outbound or Hazmat.\n",
    "+ Remember to seal your package thoroughly with tape. I recommend using [Lab Labelling Tape](https://www.fishersci.com/shop/products/fisherbrand-1-in-colored-label-tapes/p-2387311) as its easy to apply and peel off.\n",
    "+ Always Ship your packages between Monday and Wednesday to avoid being left at Room Temperature over the weekend.\n",
    "+ Keep your Package at -20&deg;C before pickup.\n",
    "+ Label the Package with the Admera Quote number on the outside.\n",
    "+ Print the Shipping Form generated by Logistics, then store it inside a **Large** ziplock bag and attach to the package with Labelling tape.\n",
    "+ I recommend paying by Credit Card instead of Purchase Order. Ask for and call their accounting department number to pay.\n",
    "\n",
    "## Option 2: Submitting your Libraries to IGM at UC San Diego: \n",
    "Use this option for Higher read depths and production libraries.\n",
    "\n",
    "### Sample Submission for Tapestation D1000\n",
    "\n",
    "+ Prepare clean 1.5ml Eppendorf tubes identical in number to the eppies for each sample/split.\n",
    "+ Download and complete the [Tapestation Sample Submission Manifest](https://igm.ucsd.edu/sites/default/files/docs/Submissions/YYYY_MM_DD_PI_TapeStation_Manifest_2021.xlsx) for Tapestation D1000 screentape assay.\n",
    "+ Please follow the instructions in the manifest for labelling your eppies.\n",
    "+ You will need to submit at least 3&mu;l of each split/sample for Tapestation D1000.\n",
    "+ Follow the instructions on the [IGM website](https://igm.ucsd.edu/genomics/submission) on how and when to submit your samples.\n",
    "\n",
    "### Submitting Samples for Sequencing: \n",
    "+ Remember to include 50&mu;l of Custom Read 1 Primer with your Submission.\n",
    "+ IGM at UCSD generally requires your samples be pooled before submission for sequencing. \n",
    "+ Please follow the instructions in the [Sequencing Manifest](https://igm.ucsd.edu/genomics/submission) for preparing your library for submission.\n",
    "    + [Here is an example](Example_Sequencing_Manifest.xlsx) Sequencing Manifest for IGM.\n",
    "+ You will need a library concentration of at least 10nM and a minimum volume of 20&mu;l.\n",
    "+ Use the average library size from the Tapestation D1000 results to estimate the concentration of your libraries in nM.\n",
    "+ Make equimolar concentrations for all your libraries being sequenced together on the same lane. For two libraries being pooled together 10&mu;l of each mixed at 10nM will lead to approximately equal reads for both libraries. You can change the fraction of each library in the final volume higher or lower depending on how many sequencing reads you want for each.\n",
    "+ You can use the code below to calculate the volume and concentration for your libraries. This code is an implementation of the [formula used by Illumina](https://knowledge.illumina.com/library-preparation/dna-library-prep/library-preparation-dna-library-prep-reference_material-list/000001240).\n",
    "$$\n",
    "Concentration\\ in\\ nM = \\frac{Concentration\\ in\\ ng/\\mu l}{660g/mol\\times Average\\ size\\ of\\ tagmented\\ library\\ in\\ bp}\\times 10^6\n",
    "$$\n",
    "\n",
    "<style>\n",
    "    @media print {\n",
    "    a[href]::after {\n",
    "        content: none !important;\n",
    "        }\n",
    "    }\n",
    "</style>\n",
    "<a id=\"nm_calculator\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Concentration in nM is: 55.322966507177036\n",
      "For 10 nM we need to dissolve 1.81 ul in  10 ul. Water needed is: 8.19 ul\n"
     ]
    }
   ],
   "source": [
    "sample_conc_ng_ul=22.2 # As measured by Qubit\n",
    "sample_average_size=608 #determined from Tapestation D1000 Results\n",
    "volume_ul=10 #Final Volume for this library.\n",
    "final_conc=10 #in nM\n",
    "\n",
    "ans=1000000*sample_conc_ng_ul/(660*sample_average_size)\n",
    "print(\"Concentration in nM is:\",ans)\n",
    "volume_needed=(volume_ul*final_conc)/ans\n",
    "print(\"For\",final_conc,\"nM we need to dissolve\",str(round(volume_needed,2)), \"ul in \",volume_ul,\"ul. Water needed is:\",str(round(volume_ul-volume_needed,2)),\"ul\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Data Analysis\n",
    "\n",
    "The [ChronoSeq Tools Repository on Github](https://github.com/kanishkasthana/ChronoSeq-Tools) has instructions and resources for Analysing your Data."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
