{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "643ec1c4",
   "metadata": {},
   "source": [
    "This Protocol is no longer in Use. See the Bead Modification for [Single Cell](protocol_for_single_cell_scale_up_chronoseqv4_dropseq_bead_modification.ipynb) or [Bulk Protocols](protocol_for_chronoseqv4_bead_modification_for_bulk_time_series.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c4911f1",
   "metadata": {},
   "source": [
    "## Test Protocol for Adding Time-Tag and PolyT modifications to ChronoSeq V3 and Dropseq beads (100&mu;M/10&mu;M Version)\n",
    "\n",
    "### ChronoSeqV3 Beads Properties\n",
    "\n",
    "#### Complete ChronoSeq V3 Sequences with Sticker Sequences:\n",
    "Beads were ordered from [Chemgenes](https://www.chemgenes.com/contact.php) and each SEQ was synthesized at a 10&mu;mol Starting Scale. <br>\n",
    "+ SEQ 1: 5’ –Bead–Linker-‐TTTTTTTAAGCAGTGGTATCAACGCAGAGTACJJJJJJJJJJJJ**ACTACGCCATCATC**-‐3’\n",
    "+ SEQ 2: 5’ –Bead–Linker-‐TTTTTTTAAGCAGTGGTATCAACGCAGAGTACJJJJJJJJJJJJ**TGATGCGGTAGTAG**-‐3’\n",
    "+ SEQ 3: 5’ –Bead–Linker-‐TTTTTTTAAGCAGTGGTATCAACGCAGAGTACJJJJJJJJJJJJ**GAGCATTTGCAGCA**-‐3’\n",
    "+ SEQ 4: 5’ –Bead–Linker-‐TTTTTTTAAGCAGTGGTATCAACGCAGAGTACJJJJJJJJJJJJ**CTCGTAAACGTCGT**-‐3’\n",
    "\n",
    "\n",
    "\n",
    "#### Sticker Sequence Properties\n",
    "\n",
    "+ Keeping Salt Concentration 50mM these are Chronoseq V3 Sticker sequence properties generated using the [Multiple Primer Analyzer Tool](https://www.thermofisher.com/us/en/home/brands/thermo-scientific/molecular-biology/molecular-biology-learning-center/molecular-biology-resource-library/thermo-scientific-web-tools/multiple-primer-analyzer.html).\n",
    "+ Pairwise Hamming distance between all the Sticker sequences is 14.\n",
    "\n",
    "|Name|Sticker Sequence|Melting Temperature°C|CG%|nt|A|T|C|G|Extinction coefficient(l/(mol·cm)|Molecular weight(g/mol)|nmol|µg/OD260|\n",
    "|-|-|-|-|-|-|-|-|-|-|-|-|-|\n",
    "|seq_1|actacgccatcatc|46.5|50.0|14|4.0|3.0|6.0|1.0|130800.0                         |4167.8                 |7.6 |31.9|\n",
    "|seq_2|tgatgcggtagtag|44.9|50.0|14|3.0|4.0|1.0|6.0|142800.0                         |4358.9                 |7.0 |30.5|\n",
    "|seq_3|gagcatttgcagca|53.3|50.0|14|4.0|3.0|3.0|4.0|138000.0                         |4287.9                 |7.2 |31.1|\n",
    "|seq_4|ctcgtaaacgtcgt|47.3|50.0|14|3.0|4.0|4.0|3.0|132000.0                         |4238.8                 |7.6 |32.1|\n",
    "\n",
    "#### Sticker Oligos for ChronoSeq V3 Seq 1:\n",
    "\n",
    "> All Oligos were ordered from [IDT](https://www.idtdna.com/site/order/oligoentry) \n",
    "\n",
    "+ **SEQ 1 Barcode 1 HPLC Oligo**: Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA CTCTCTCT **GATGATGGCGTAGT**\n",
    "    + This Sequence was HPLC Purified.\n",
    "    + The bold sequence is the reverse complement of the ChronoSeq V3 Seq 1 Sticker Sequence.\n",
    "    + The Sequence in the middle is a Test Time Tag\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "+ **3Am_ChronoV3_S1B1**: Sequence: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA<em>NNNNCNCNCNCN</em>**GATGATGGCGTAGT**/3AmMO/\n",
    "    + Standard Desalting was used for this Sequence.\n",
    "    + The bold sequence is the reverse complement of the ChronoSeq V3 Seq 1 Sticker Sequence.\n",
    "    + The Sequences in Italics is a Combined UMI and Time-Tag. The C Bases are currently used for the time-tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "    + The 3' Amino Modification is for Blocking DNA extension. \n",
    "    \n",
    "### Dropseq beads Properties and Sticker Sequence\n",
    "5’–Bead–Linker-‐TTTTTTTAAGCAGTGGTATCAACGCAGAGTAC JJJJJJJJJJJJNNNNNNNN TTTTTTTTTTTTTTTTTTTTTTTTTTTTTT-‐3’ \n",
    "Beads were ordered from [Chemgenes](https://www.chemgenes.com/contact.php) and synthesized at a 10&mu;mol Starting Scale.\n",
    "\n",
    "#### Sticker Oligos for Dropseq beads\n",
    "\n",
    "> All Oligos were ordered from [IDT](https://www.idtdna.com/site/order/oligoentry) \n",
    "\n",
    "+ **A30N7BA19 HPLC Oligo**: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA NNNNNNNB AAAAAAAAAAAAAAAAAAA\n",
    "    + This Sequence was HPLC Purified when ordering from IDT.\n",
    "    + The A19 Part of the Sequence is for binding the PolyT part of the Dropseq Beads.\n",
    "    + The N7B is the test Time Tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "+ **3Am_Drop_B1**: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA **GNGNGNGB** AAAAAAAAAAAAAAAAAAA/3AmMO/\n",
    "    + Standard Desalting was used for this Sequence.\n",
    "    + The A19 Part of the Sequence is for binding the PolyT part of the Dropseq Beads.\n",
    "    + The Sequences in Bold is a Time-Tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "    + The 3' Amino Modification is for Blocking DNA extension.\n",
    "    \n",
    "#### After Looking at Sequencing Data from First Round of Experiments Testing Important Combinations:\n",
    "+ **DTag_TTGG_CCAA**: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA **TNTHGHGB** AAAAAAAAAAAAAAAAAAA\n",
    "    + Standard Desalting was used for this Sequence.\n",
    "    + The A19 Part of the Sequence is for binding the PolyT part of the Dropseq Beads.\n",
    "    + The Sequences in Bold is a Time-Tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "+ **DTag_CCTT_AAGG**: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA **CDCDTNTB** AAAAAAAAAAAAAAAAAAA\n",
    "    + Standard Desalting was used for this Sequence.\n",
    "    + The A19 Part of the Sequence is for binding the PolyT part of the Dropseq Beads.\n",
    "    + The Sequences in Bold is a Time-Tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "+ **DTag_GGAA_TTCC**: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA **GHGHANAB** AAAAAAAAAAAAAAAAAAA\n",
    "    + Standard Desalting was used for this Sequence.\n",
    "    + The A19 Part of the Sequence is for binding the PolyT part of the Dropseq Beads.\n",
    "    + The Sequences in Bold is a Time-Tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "+ **DTag_TTCC_GGAA**: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA **TNTDCDCB** AAAAAAAAAAAAAAAAAAA\n",
    "    + Standard Desalting was used for this Sequence.\n",
    "    + The A19 Part of the Sequence is for binding the PolyT part of the Dropseq Beads.\n",
    "    + The Sequences in Bold is a Time-Tag.\n",
    "    + The PolyA sequence is for making a PolyT sequence on the Beads.\n",
    "\n",
    "### Calculations and Notes:\n",
    "\n",
    "+ The scale of synthesis for each SEQ or Batch of Dropseq Beads is 10&mu;mol.\n",
    "+ We will divide these beads into 1/4 parts for each single Time-tag. This is assuming we want 16 Time-Tags from SEQ1-SE4 ChronoSeq V3 or 4 from 10&mu;mol Drop-Seq beads.\n",
    "+ So about 2.5&mu;mol per Time-tag. 10&mu;mol of Beads comes to about 12-13mls at 450beads/&mu;l. This comes to about 3 mls of beads at 450beads/&mu;l for each 1/4 Split for a Time-Tag.\n",
    "+ We have previously done an [experiment](protocol_for_exo1_predigestion_for_chronoseqV2_and_dropseq_beads.ipynb) with 0.4ml of beads at 450beads/&mu;l. This should be equivalent to about 0.25&mu;mol.\n",
    "+ This previous experiment we had no problems when we did the Exo 1 Digestion.\n",
    "+ We have several things to keep in mind as constraints for our future experiments now.\n",
    "+ If we do the whole 2.5&mu;mols we should expect to have about 1.25&mu;mol of complete primers and 2.5&mu;mols including incomplete primers. Incomplete primers will be relevant for Exonuclease 1 Digestion.\n",
    "+ So if we assume we will be synthesizing 100bases for each complete sequence we need 125&mu;mols of total dNTPs. But since the reaction mixture is equimolar and we need about half our bases to be dTs we need about half or 50&mu;mols of dNTPs.**Thats a lot. Too much infact.**\n",
    "+ 6&mu;mols for the 0.4ml fraction at least. But we used a much smaller number of primers in our previous experiment and this did not create any problems (Exo1 + Alkaline Denaturation) compared to unmodified Drop-Seq. So from now onwards we will use the primer molar amount for our calculations as a baseline.\n",
    "\n",
    "#### Calculations for 0.25&mu;mol/0.4ml Test Batch:\n",
    "\n",
    "+ So for our test reaction with 0.25&mu;mol.\n",
    "+ 20&mu;l of 1mM primers is 10nmol of primers.\n",
    "+ So for 100 bases we need about 1000 nmol of dNTPs or 500nmol for dTs in particular. T(30) + 70/4. so 0.5&mu;mol total of eqimolar primers. 100&mu;l of [dNTP mix](https://www.neb.com/en-us/products/n0447-deoxynucleotide-dntp-solution-mix#Product%20Information) should be enough for that. We can titrate this down if we see equally good results for lower amounts of primers. Just multiply the primer amount by 100 and that should give us the amount of dNTPs needed. \n",
    "+ We also need a Polymerase that is fast enough to get this done and do it at a temperature lower than the melting temperature of our ChronoSeq V3 beads.\n",
    "+ phi29 DNA polymerase produces blunt ends but is too slow. [Only 0.5pmol of dNTPs at 30&deg;C in 10 minutes.](https://www.neb.com/en-us/products/m0269-phi29-dna-polymerase#Product%20Information_Properties%20&%20Usage)\n",
    "+ We might use Bsu DNA polymerase. The optimum reaction temperature for this polymerase is lower than our sticker sequence melting temperatures at 37&deg;C. Moreover the enzyme keeps 50% of its activity at 25&deg;C which is still pretty fast. However, this enzyme produces [3'A single base overhangs](https://www.neb.com/en-us/products/m0330-bsu-dna-polymerase-large-fragment#Product%20Information_Properties%20&%20Usage) which can cause problems later.\n",
    "+ [DNA Polymerase I](https://www.neb.com/en-us/products/m0209-dna-polymerase-i-e-coli#Product%20Information) produces blunt ends. Moreover, the optimum reaction temperature for this polymerase is lower than our sticker sequence melting temperatures at 37&deg;C. However, we need to be careful in deciding the reaction conditions to avoid getting recessed ends.\n",
    "+ According to the [NEB Website](https://www.neb.com/en-us/products/m0209-dna-polymerase-i-e-coli#Product%20Information_Properties%20&%20Usage) at 37&deg;C we get 10nmol of dNTPs incoporated by one unit of DNA Polymerase I in 30 minutes. There are 10units/&mu;l and we will use 5&mu;l of polymerase. So that gives us 500nmol of dNTPs per 30 minutes. This reaction should be complete in 1 hour at 37&deg;C. We might need to increase this time for larger amount of primer or beads in our reaction mixture in the future. \n",
    "+ Lets run this reaction at 34&deg;C assuming 75% activity levels. We will run this reaction for 1.5 hours. This should be enough time to complete the reaction and [avoid recessed ends](https://www.neb.com/en-us/products/m0210-dna-polymerase-i-large-klenow-fragment#Product%20Information_Product%20Notes). We assume the Polymerase will have 25% Activity levels similar to the DNA Polymerase I, Large Klenow Fragment since [direct information for this Polymerase is not available](https://www.neb.com/en-us/products/m0330-bsu-dna-polymerase-large-fragment#Product%20Information_Product%20Notes). 100% at 37&deg;C and 25% at 25&deg;C gives 75% Activity at 34&deg;C using a linear approximation. \n",
    "+ We will then run an [Exonuclease 1](https://www.neb.com/en-us/products/m0293-exonuclease-i-e-coli#Product%20Information) to get rid of the primers that were not extended.\n",
    "+ According to the [NEB website](https://www.neb.com/en-us/products/m0293-exonuclease-i-e-coli#Product%20Information_Properties%20&%20Usage) 1 unit of Exonuclease 1 removed 10nmol at 37&deg;C in 30 minutes. There are 20 units/&mu;l and we will use 50&mu;l for our reaction. So we can remove 20&mu;mol of dNTPs in 1 hour. For 2.5&mu;mol reaction we need to remove about 56-82 bases depending on if its Dropseq or Chronoseq V3. 210&mu;mol for Dropseq considering only a small fraction of sequences will actually have the primer. We will have to run this reaciton for 11 hours in that case or we could use more enzyme. But for the 0.25&mu;mol reaction we need to remove at most 21&mu;mol. So in 1.5 hours the reaction should be complete at 37&deg;C. So our previous attempt should have completely gotten rid of these single stranded primers.\n",
    "+ After this step we need to get rid of the primer through Alkaline denaturation similar to before with the Exonuclease 1 and Alkaline denaturation protocol. We also need to maintain the salt concentration to make sure the primers don't get removed before the Alkaline Denaturation step. We want the Exonuclease 1 to act only on ssDNA and not dsDNA which is why this step is important.\n",
    "\n",
    "### Solutions and Buffers to Make\n",
    "Use [Distilled Water](https://www.fishersci.com/shop/products/invitrogen-ultrapure-dnase-rnase-free-distilled-water-2/10977023) for all these Buffers. Please also filter using a [40&mu;m Cell Strainer](https://www.fishersci.com/shop/products/falcon-cell-strainers-4/087711).\n",
    "+ STOP Buffer:\n",
    "    + 10 mM [Tris HCl pH 8.0](https://www.fishersci.com/shop/products/1m-tris-hcl-ph-8-0-1l/50843267) + 1 mM [EDTA](https://www.fishersci.com/shop/products/ambion-edta-0-5m-ph-8-0-3/am9261)\n",
    "    + 0.5% [Sarcosyl](https://www.sigmaaldrich.com/US/en/product/sigma/l7414)\n",
    "    + 330mM [KCl](https://www.fishersci.com/shop/products/ambion-kcl-2m/AM9640G)\n",
    "+ WASH Buffer:\n",
    "    + 10 mM Tris HCl pH 8.0 + 1 mM EDTA\n",
    "    + 0.01% [Tween-20](https://www.gbiosciences.com/Protein-Research/Tween-20) \n",
    "    + 330mM KCl\n",
    "+ TE-SDS:\n",
    "    + 10 mM [Tris HCl pH 8.0](https://www.fishersci.com/shop/products/1m-tris-hcl-ph-8-0-1l/50843267) + 1 mM [EDTA](https://www.fishersci.com/shop/products/ambion-edta-0-5m-ph-8-0-3/am9261)\n",
    "    + 0.5% [SDS](https://www.fishersci.com/shop/products/sds-solution-20-sodium-dodecyl-sulfate-solution-molecular-biology-electrophoresis-fisher-bioreagents/BP1311200)\n",
    "+ TE-TW:\n",
    "    + 10 mM Tris HCl pH 8.0 + 1 mM EDTA\n",
    "    + 0.01% [Tween-20](https://www.gbiosciences.com/Protein-Research/Tween-20)\n",
    "\n",
    "+ IDTE Buffer:\n",
    "    + 10 mM Tris HCl pH 8.0 + 0.1mM EDTA\n",
    "    \n",
    "> Buffer for Dissolving and Diluting Oligos Ordered from IDT    \n",
    "\n",
    "+ 1M DTT:\n",
    "    + Dissolve 1.55g solid [DTT](https://www.fishersci.com/shop/products/dithiothreitol-white-crystals-powder-electrophoresis-fisher-bioreagents/BP17225) in 10ml of [Deionized Water](https://www.fishersci.com/shop/products/water-reagent-deionized-water-acs-thermo-scientific/AA36645K2). **Hazardous Chemical [Read SDS](https://www.fishersci.com/store/msds?partNumber=BP17225&productDescription=DTT+25G&vendorId=VN00033897&countryCode=US&language=en). Work in Fume Hood** \n",
    "\n",
    "> More economical to make yourself from Solid DTT. Make 50ml or more at a time and store in 10ml Aliquots for upto an year.  DTT is not stable at room temperature. Make this solution quickly and store at -20&deg;C, unless you plan to use it immediately to make another Buffer. \n",
    "\n",
    "+ 10X NEB Buffer 2:\n",
    "    + 500 mM [NaCl](https://www.fishersci.com/shop/products/ambion-nacl-5m-2/AM9760G)\n",
    "    + 100 mM [Tris-HCl pH 7.9](https://www.fishersci.com/shop/products/tris-hcl-buffer-1m-ph-7-9-st/NC2212672)\n",
    "    + 100 mM [MgCl<sub>2</sub>](https://www.fishersci.com/shop/products/ambion-mgcl-sub-2-sub-1m/AM9530G)\n",
    "    + 10 mM DTT\n",
    "\n",
    "> Make in Bulk for Scaling up reaction. It is more economical than buying [NEB Buffer 2](https://www.neb.com/en-us/products/b7002-nebuffer-2#Product%20Information). Make 100ml or more at a time and store in 11ml Aliquots for upto an year.\n",
    "DTT is not stable at room temperature. Make this solution quickly and store at -20&deg;C. \n",
    "\n",
    "\n",
    "### Protocol for Bead Modification\n",
    "#### Preparation of Beads\n",
    "+ Make 100&mu;M/10&mu;M stocks for the Dropseq/ChronoSeqV3 Sticker Oligos ordered from IDT, using the IDTE buffer.\n",
    "+ Follow the [ChronoSeq beads preparation protocol](protocol_for_preparing_chronoseq_beads.ipynb) and suspend 1/4 part of the Beads from the Alcohol stock at -20&deg;C to [1X Lysis buffer](protocol_for_preparing_3.33X_lysis_buffer_stock.ipynb) at 4&deg;C. Do this for both ChronoSeq V3 SEQ 1 and the latest Dropseq Beads.\n",
    "+ Spin Down ChronoSeq or Dropseq beads with 450beads/&mu;l suspended in 1X Lysis buffer.\n",
    "+ Turn on the UV and airflow for the Cell Culture Hood. Leave for 15 minutes.\n",
    "+ Inside the Cell Culture Hood: Shake Each Tube to make sure the beads are <b>evenly suspended</b> . For each Tube using a <b>Sterile/Particle Free</b> [Rainin P1000 pipette tip](https://www.shoprainin.com/Pipette-Tips/Universal-Pipette-Tips/Hinged-Cover-Racks/Pipette-Tips-RT-UNV-1000%C2%B5L-LS-768A-8/p/30389169) transfer 400&mu;l of beads to a <b>Sterile</b> 1.5ml Eppendorf Tube.\n",
    "+ Put the Tubes back in their storage location at 4&deg;C once you are done.\n",
    "+ Turn off the Cell Culture Hood and move to your bench.\n",
    "\n",
    "> This is a **STOPPING STEP**. You can store the Beads in 1X Lysis Buffer at 4&deg;C and continue later.\n",
    "\n",
    "#### DNA Polymerase I Isothermal Extension\n",
    "+ Get ChronoSeq V3 or new Dropseq beads suspended in 1X Lysis buffer.\n",
    "+ Set an incubator to 34&deg;C\n",
    "+ Get the dNTP, Sticker Oligos and [NEB Buffer 2](https://www.neb.com/en-us/products/b7002-nebuffer-2#Product%20Information) from -20&deg;C and leave on your bench to melt. Put on ice once melted.\n",
    "> &#128161; **Tip:** Make sure you have enough volume of dNTPs, Oligos and Buffer for your experiments.\n",
    "+ Get the DNA Polymerase I from -20&deg;C and put it on ice.\n",
    "+ Make 1.25X NEB Buffer 2 and put on ice.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Remove 300&mu;l of Lysis buffer without disturbing the beads.\n",
    "+ Now wash the beads 3 Times with 1ml 1.25X NEB Buffer 2:\n",
    "    + Add 1ml 1.25X NEB Buffer 2. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml 1.25X NEB Buffer 2. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml 1.25X NEB Buffer 2. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Without disturbing the beads remove 400&mu;l of Buffer.\n",
    "+ Estimate the Volume using the P1000.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Remove liquid without distrubing the beads to make the final Volume 320&mu;l.\n",
    "+ Now add:\n",
    "    + 20&mu;l of 10&mu;M Sticker Oligo for Dropseq/ChronoSeq-V3. Pipette up and down several times to mix well.\n",
    "    + 20&mu;l dNTP mix from NEB\n",
    "    + 39&mu;l [Distilled Water](https://www.fishersci.com/shop/products/invitrogen-ultrapure-dnase-rnase-free-distilled-water-2/10977023)\n",
    "    + 1&mu;l DNA Polymerase I . <em>Use the P2.5 , Pipette up and down several times to mix well.</em>\n",
    "+ Vortex the Beads at full speed for 2 seconds to mix everything well.\n",
    "+ Now Incubate at 34&deg;C with Rotation for 1.5 hours.\n",
    "+ Set the incubator to 37&deg;C after you are done using it.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Without disturbing the beads remove 200&mu;l of Buffer.\n",
    "+ Wash the beads once with STOP Buffer and then twice with WASH Buffer:\n",
    "    + Add 1 ml of STOP Buffer to the eppie.\n",
    "    + Flip the tube upside down a few times and then Vortex the beads at full speed for 2 seconds. \n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of WASH Buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of WASH Buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "\n",
    "> This is a **STOPPING STEP**. You can store the Beads in WASH Buffer at 4&deg;C and continue later.\n",
    "\n",
    "#### Exonuclease 1 Digestion\n",
    "+ Make sure the incubator is Set to 37&deg;C.\n",
    "+ Get the [Exonuclease 1 Buffer](https://www.neb.com/en-us/products/b0293-exonuclease-i-reaction-buffer#Product%20Information) -20&deg;C and leave on your bench to melt. Put on ice once melted.\n",
    "> &#128161; **Tip:** Make sure you have enough volume of Exonuclease 1 Buffer for your experiment.\n",
    "+ Get the Exonuclease 1 from -20&deg;C and put it on ice.\n",
    "+ Make 1.25X Exonuclease 1 Buffer and put it on ice.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Wash twice with 1ml of 1.25X Exonuclease 1 Buffer:\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1 ml of 1.25X Exo 1 buffer to the eppie. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1 ml of 1.25X Exo 1 buffer to the eppie. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute. \n",
    "+ Remove 400&mu;l of Liquid from the eppie without disturbing the beads.\n",
    "+ Without touching the beads, slightly pipette up and down using the P1000 to resuspend the beads in solution.\n",
    "+ Now estimate the Volume in the eppie by using the P1000.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Using the estimated volume in the eppie, make the final volume of the eppie 320&mu;l without disturbing the beads.\n",
    "+ Then add the following to the eppie :\n",
    "    + 50&mu;l Exonuclease 1\n",
    "    + 30&mu;l Distilled Water\n",
    "+ Vortex the Beads to Mix everything Well.\n",
    "+ Now Incubate at 37&deg;C with Rotation for 2 hours.\n",
    "+ Set the incubator to 42&deg;C after you are done using it.\n",
    "+ Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Without disturbing the beads remove 200&mu;l of Buffer.\n",
    "+ Wash the beads once with TE-SDS and then twice with TE-TW:\n",
    "    + Add 1 ml of TE-SDS to the eppie.\n",
    "    + Flip the tube upside down a few times and then Vortex the beads at full speed for 2 seconds. \n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of TE-TW. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of TE-TW. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    \n",
    "> This is a **STOPPING STEP**. You can store the Beads in TE-TW at 4&deg;C and continue later.\n",
    "\n",
    "+ Wash the beads twice with Distilled Water:\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of Distilled Water. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml of Distilled Water. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "\n",
    "### Protocol for Converting dsDNA on the Beads to ssDNA [using Alkaline Denaturation](http://dx.doi.org/10.5620/eht.2014.29.e2014007)\n",
    "+ Sodium Hydroxide is a very dangerous Chemical it can melt your Skin and turn your body into Soap. You need to be extra careful.\n",
    "+ Wear Lab Coat, Safety Glasses, [Apron](https://www.mcmaster.com/50085T13/), Face-Sheild, Double Nitrile Gloves and work in the Fume Hood. \n",
    "+ Make sure you don't have exposed skin by covering your gloves with the apron.\n",
    "+ Replace the work area mats with [Fresh new ones](https://us.vwr.com/store/product/11988983/aqui-padtm-benchtop-absorbent-mat-therapak).\n",
    "+ Make sure there is enough space to work inside the Fume hood. \n",
    "+ Bring the [10M NaOH Bottle](https://www.fishersci.com/shop/products/sodium-hydroxide-10n-aq-soln-thermo-scientific/AAJ63736AE) stored under your bench in a secondary container to the Fume Hood. You should always store the bottle inside a secondary container inside another larger container.\n",
    "+ You will need one Hazardous Waste Bin for the Tips. Bring The Black Waste Bin into The Fume Hood. Check the Waste Tag to make sure this is the correct Bin.\n",
    "> &#128161; **Tip:** Make sure you have enough volume of Denaturation Solution and 1M Tris-HCl pH 7.5 for your experiments.\n",
    "+ You will also need a Buffer to Dispose off the 1M NaOH washed through.\n",
    "+ For this purpose I recommend having between 30-40ml of [1M Tris-HCl pH 7.5](https://www.fishersci.com/shop/products/molecular-biology-reagents/MT46030CM) in a 50ml Tube Ready. Label this Tube: **Neutralization**.\n",
    "+ Set aside enough of 1M Tris-HCl pH 7.5 in a different tube for washing the Beads. Label this Tube: **Washing**.\n",
    "+ You will also need to prepare a Spray Bottle full of 1M Tris-HCl pH 7.5 Buffer. This will act as our emergency spray in case of a spill. Label This bottle correctly and Keep it inside the Fume Hood. Very important to have this with you.\n",
    "> &sext; **Important**: Discard the tips used for Handling NaOH into the Black Hazardous Waste Bin.\n",
    "> <br>**In case of a Spill**: Spray all affected surfaces with the 1M Tris-HCl pH 7.5.\n",
    "+ We Will now make Fresh Denaturation Solution. To Make 1ml of Denaturation Solution:\n",
    "    + 900&mu;l of Distilled Water\n",
    "    + 100&mu;l of 10N NaOH.\n",
    "    + Pipette up and down to mix well.\n",
    "    + Discard the NaOH contaminated tip into the Black Bin.\n",
    "+ Once you have made enough Denaturation Solution, put the 10M NaOH back in its storage location inside its secondary container. Make sure the 10M NaOH bottle is closed tightly.\n",
    "+ Change your gloves.\n",
    "+ Now wash the beads thrice with 1ml Denaturation Solution:\n",
    "    + Now Carefully Add 1ml of Denaturation Solution to the Beads. Discard the Tip into the Black Bin.\n",
    "    + Vortex the Beads at Maximum Speed for 3 seconds to Mix Well. \n",
    "    + Incubate at Room temperature with Rotation for 5 minutes.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads. Discard this liquid into the **Neutralization** Tube.\n",
    "    + Now Discard the Tip into the Black Bin.\n",
    "    + Now Carefully Add 1ml of Denaturation Solution to the Beads. Discard the Tip into the Black Bin.\n",
    "    + Vortex the Beads at Maximum Speed for 3 seconds to Mix Well.\n",
    "    + Incubate at Room temperature with Rotation for 5 minutes.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads. Discard this liquid into the **Neutralization** Tube.\n",
    "    + Now Discard the Tip into the Black Bin.\n",
    "    + Now Carefully Add 1ml of Denaturation Solution to the Beads. Discard the Tip into the Black Bin.\n",
    "    + Vortex the Beads at Maximum Speed for 3 seconds to Mix Well.\n",
    "    + Incubate at Room temperature with Rotation for 5 minutes.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads. Discard this liquid into the **Neutralization** Tube.\n",
    "    + Now Discard the Tip into the Black Bin.\n",
    "+ Now wash Twice with 1ml 1M Tris-HCl pH 7.5:\n",
    "    + Now Carefully Add 1ml of 1M Tris-HCl pH 7.5 from the **Washing** tube to the Beads. Discard the Tip into the Black Bin.\n",
    "    + Vortex the Beads at Maximum Speed for 3 seconds to Mix Well.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads. Discard this liquid into the **Neutralization** Tube.\n",
    "    + Now Discard the Tip into the Black Bin.\n",
    "    + Now Carefully Add 1ml of 1M Tris-HCl pH 7.5 from the **Washing** tube to the Beads. Discard the Tip into the Black Bin.\n",
    "    + Vortex the Beads at Maximum Speed for 3 seconds to Mix Well.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads. Discard this liquid into the **Neutralization** Tube.\n",
    "+ Store the Black bin, Apron, Spray Bottle and other items in their correct location.\n",
    "+ Use [pH test Paper](https://www.fishersci.com/shop/products/hydrion-ph-test-paper-6/14850120) to check the pH of the Neutralization Tube. If the pH is less than 10 you can pour it in the sink. Additional Dilution with 1M Tris-HCl pH 7.5 might be needed if its greater in pH.\n",
    "+ You can now move back to your bench from the fume hood.\n",
    "+ Wash the beads thrice with 1ml 1X Lysis Buffer:\n",
    "    + Add 1ml 1X Lysis Buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml 1X Lysis Buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Carefully remove 1ml of Liquid without disturbing the beads.\n",
    "    + Add 1ml 1X Lysis Buffer. Vortex the beads at full speed for 2 seconds to resuspend the beads if necessary.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "+ We now need to make a Final concentration of 450beads/&mu;l:\n",
    "    + Remove 300&mu;l of Lysis buffer.\n",
    "    + Vortex the beads at Maximum Speed for 3 seconds and then quickly take a 20&mu;l Sample using the P200. Without delay transfer this sample to a [Fuchs-Rosenthal Hemocytometer](https://www.fishersci.com/shop/products/c-chip-disposable-hemacytometers-2/22600102) \n",
    "    + Count the Beads to get the Bead Concentration. Count at least 5 large squares to get the bead concentration per &mu;l\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Estimate the Volume of the Beads using a P1000.\n",
    "    + Spin down the beads using a tabletop centrifuge @ 2500xg for 1 minute.\n",
    "    + Without disturbing the beads, remove the calculated amount of liquid to get a Final Concentration of 450beads/&mu;l.\n",
    "> This is a **STOPPING STEP**. You can store the Beads in 1X Lysis Buffer at 4&deg;C and continue later.\n",
    "+ Follow the [Bulk RNA Seq protocol](protocol_for_bulk_rna_seq_library_prep.ipynb) to verify your beads are working or not.\n",
    "<style>\n",
    "    @media print {\n",
    "    a[href]::after {\n",
    "        content: none !important;\n",
    "        }\n",
    "    }\n",
    "</style>"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
